{"cells":[{"metadata":{},"cell_type":"markdown","source":"***Import Packages***"},{"metadata":{"_uuid":"91534c55-a219-4d94-b60d-f0d18fd8445a","_cell_guid":"8c3cf103-ae5a-446b-bfa5-7e7a1f726ed2","trusted":true},"cell_type":"code","source":"import numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\nimport glob\nimport json\nimport os\nimport matplotlib.pyplot as plt\nplt.style.use('ggplot')","execution_count":1,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"***Load Dataset***"},{"metadata":{"trusted":true},"cell_type":"code","source":"biorxiv_dir = '/kaggle/input/CORD-19-research-challenge/biorxiv_medrxiv/biorxiv_medrxiv/pdf_json'\nfilenames = os.listdir(biorxiv_dir)\nprint(\"Number of articles retrieved from biorxiv:\", len(filenames))","execution_count":2,"outputs":[{"output_type":"stream","text":"Number of articles retrieved from biorxiv: 1342\n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"***Helper Functions***"},{"metadata":{"trusted":true},"cell_type":"code","source":"def load_files(dirname):\n    filenames = os.listdir(dirname)\n    raw_files = []\n\n    for filename in tqdm(filenames):\n        filename = dirname + filename\n        file = json.load(open(filename, 'rb'))\n        raw_files.append(file)\n    \n    return raw_files\n\ndef generate_clean_df(all_files):\n    cleaned_files = []\n    \n    for file in tqdm(all_files):\n        features = [\n            file['paper_id'],\n            file['metadata']['title'],\n            format_authors(file['metadata']['authors']),\n            format_authors(file['metadata']['authors'], \n                           with_affiliation=True),\n            format_body(file['abstract']),\n            format_body(file['body_text']),\n            format_bib(file['bib_entries']),\n            file['metadata']['authors'],\n            file['bib_entries']\n        ]\n\n        cleaned_files.append(features)\n\n    col_names = ['paper_id', 'title', 'authors',\n                 'affiliations', 'abstract', 'text', \n                 'bibliography','raw_authors','raw_bibliography']\n\n    clean_df = pd.DataFrame(cleaned_files, columns=col_names)\n    clean_df.head()\n    \n    return clean_df","execution_count":3,"outputs":[]},{"metadata":{},"cell_type":"markdown","source":"***Helper Functions***"},{"metadata":{"trusted":true},"cell_type":"code","source":"def format_name(author):\n    middle_name = \" \".join(author['middle'])\n    \n    if author['middle']:\n        return \" \".join([author['first'], middle_name, author['last']])\n    else:\n        return \" \".join([author['first'], author['last']])\n\n\ndef format_affiliation(affiliation):\n    text = []\n    location = affiliation.get('location')\n    if location:\n        text.extend(list(affiliation['location'].values()))\n    \n    institution = affiliation.get('institution')\n    if institution:\n        text = [institution] + text\n    return \", \".join(text)\n\ndef format_authors(authors, with_affiliation=False):\n    name_ls = []\n    \n    for author in authors:\n        name = format_name(author)\n        if with_affiliation:\n            affiliation = format_affiliation(author['affiliation'])\n            if affiliation:\n                name_ls.append(f\"{name} ({affiliation})\")\n            else:\n                name_ls.append(name)\n        else:\n            name_ls.append(name)\n    \n    return \", \".join(name_ls)\n\ndef format_body(body_text):\n    texts = [(di['section'], di['text']) for di in body_text]\n    texts_di = {di['section']: \"\" for di in body_text}\n    \n    for section, text in texts:\n        texts_di[section] += text\n\n    body = \"\"\n\n    for section, text in texts_di.items():\n        body += section\n        body += \"\\n\\n\"\n        body += text\n        body += \"\\n\\n\"\n    \n    return body\n\ndef format_bib(bibs):\n    if type(bibs) == dict:\n        bibs = list(bibs.values())\n    bibs = deepcopy(bibs)\n    formatted = []\n    \n    for bib in bibs:\n        bib['authors'] = format_authors(\n            bib['authors'], \n            with_affiliation=False\n        )\n        formatted_ls = [str(bib[k]) for k in ['title', 'authors', 'venue', 'year']]\n        formatted.append(\", \".join(formatted_ls))\n\n    return \"; \".join(formatted)","execution_count":4,"outputs":[]},{"metadata":{"_uuid":"6dec0274-7a13-4faf-8c9b-ee73f5949ea5","_cell_guid":"8415a09d-49f4-471c-a426-2fa9b839a949","trusted":true},"cell_type":"code","source":"all_files = []\n\nfor filename in filenames:\n    filename = biorxiv_dir +\"/\"+ filename\n    file =json.load(open(filename, 'rb'))\n    all_files.append(file)\n\n    \n\n","execution_count":5,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"def format_name(author):\n    middle_name = \" \".join(author['middle'])\n    \n    if author['middle']:\n        return \" \".join([author['first'], middle_name, author['last']])\n    else:\n        return \" \".join([author['first'], author['last']])\n\n\ndef format_affiliation(affiliation):\n    text = []\n    location = affiliation.get('location')\n    if location:\n        text.extend(list(affiliation['location'].values()))\n    \n    institution = affiliation.get('institution')\n    if institution:\n        text = [institution] + text\n    return \", \".join(text)\n\ndef format_authors(authors, with_affiliation=False):\n    name_ls = []\n    \n    for author in authors:\n        name = format_name(author)\n        if with_affiliation:\n            affiliation = format_affiliation(author['affiliation'])\n            if affiliation:\n                name_ls.append(f\"{name} ({affiliation})\")\n            else:\n                name_ls.append(name)\n        else:\n            name_ls.append(name)\n    \n    return \", \".join(name_ls)\n\ndef format_body(body_text):\n    texts = [(di['section'], di['text']) for di in body_text]\n    texts_di = {di['section']: \"\" for di in body_text}\n    \n    for section, text in texts:\n        texts_di[section] += text\n\n    body = \"\"\n\n    for section, text in texts_di.items():\n        body += section\n        body += \"\\n\\n\"\n        body += text\n        body += \"\\n\\n\"\n    \n    return body\n","execution_count":6,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"from tqdm.notebook import tqdm\n\ncleaned_files = []\n    \nfor file in tqdm(all_files):\n    features = [\n        file['paper_id'],\n        file['metadata']['title'],\n        format_authors(file['metadata']['authors']),\n        format_authors(file['metadata']['authors'], \n                       with_affiliation=True),\n        format_body(file['abstract']),\n        format_body(file['body_text']),\n\n    ]\n\n    cleaned_files.append(features)\n\ncol_names = ['paper_id', 'title', 'authors',\n             'affiliations', 'abstract', 'text',]\n\ndf = pd.DataFrame(cleaned_files, columns=col_names)\ndf.head()","execution_count":7,"outputs":[{"output_type":"display_data","data":{"text/plain":"HBox(children=(FloatProgress(value=0.0, max=1342.0), HTML(value='')))","application/vnd.jupyter.widget-view+json":{"version_major":2,"version_minor":0,"model_id":"871fc334936441ba94b9a5b38e1e4f9f"}},"metadata":{}},{"output_type":"stream","text":"\n","name":"stdout"},{"output_type":"execute_result","execution_count":7,"data":{"text/plain":"                                   paper_id  \\\n0  4602afcb8d95ebd9da583124384fd74299d20f5b   \n1  90b5ecf991032f3918ad43b252e17d1171b4ea63   \n2  c8a9e49416051620631005def6ea86c91cf86bc9   \n3  d3c2e2839498c613ee95739dce7052109750362c   \n4  bbd9d63dc2c733c763770f62205ef9adeceb0127   \n\n                                               title  \\\n0  SPINT2 inhibits proteases involved in activati...   \n1  The role of absolute humidity on transmission ...   \n2  FEP-based screening prompts drug repositioning...   \n3  Long-Term Persistence of IgG Antibodies in SAR...   \n4  Effects of temperature variation and humidity ...   \n\n                                             authors  \\\n0  Marco R Straus, Jonathan T Kinder, Michal Sega...   \n1  Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...   \n2  Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...   \n3  Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...   \n4  Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...   \n\n                                        affiliations  \\\n0  Marco R Straus, Jonathan T Kinder (University ...   \n1  Wei Luo (Boston Children's Hospital, 02215, Bo...   \n2  Zhe Li (Sun Yat-Sen University, 510006, Guangz...   \n3  Xiaoqin Guo (Sun Yat-sen University, 510080, G...   \n4  Yueling Ma (Lanzhou University, 730000, Lanzho...   \n\n                                            abstract  \\\n0  Abstract\\n\\nViruses possessing class I fusion ...   \n1                                                      \n2  Abstract\\n\\nThe new coronavirus COVID-19, also...   \n3  Abstract\\n\\n23 BACKGROUND 24 The ongoing world...   \n4  Abstract\\n\\nObject Meteorological parameters a...   \n\n                                                text  \n0  Introduction 9\\n\\nInfluenza-like illnesses (IL...  \n1  Introduction\\n\\nSince December 2019, an increa...  \n2  Introduction\\n\\nThe novel coronavirus 2019-nCo...  \n3  \\n\\nCC-BY-ND 4.0 International license It is m...  \n4  Introduction\\n\\nIn December 2019, a novel coro...  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>paper_id</th>\n      <th>title</th>\n      <th>authors</th>\n      <th>affiliations</th>\n      <th>abstract</th>\n      <th>text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>4602afcb8d95ebd9da583124384fd74299d20f5b</td>\n      <td>SPINT2 inhibits proteases involved in activati...</td>\n      <td>Marco R Straus, Jonathan T Kinder, Michal Sega...</td>\n      <td>Marco R Straus, Jonathan T Kinder (University ...</td>\n      <td>Abstract\\n\\nViruses possessing class I fusion ...</td>\n      <td>Introduction 9\\n\\nInfluenza-like illnesses (IL...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>90b5ecf991032f3918ad43b252e17d1171b4ea63</td>\n      <td>The role of absolute humidity on transmission ...</td>\n      <td>Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...</td>\n      <td>Wei Luo (Boston Children's Hospital, 02215, Bo...</td>\n      <td></td>\n      <td>Introduction\\n\\nSince December 2019, an increa...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>c8a9e49416051620631005def6ea86c91cf86bc9</td>\n      <td>FEP-based screening prompts drug repositioning...</td>\n      <td>Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...</td>\n      <td>Zhe Li (Sun Yat-Sen University, 510006, Guangz...</td>\n      <td>Abstract\\n\\nThe new coronavirus COVID-19, also...</td>\n      <td>Introduction\\n\\nThe novel coronavirus 2019-nCo...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>d3c2e2839498c613ee95739dce7052109750362c</td>\n      <td>Long-Term Persistence of IgG Antibodies in SAR...</td>\n      <td>Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...</td>\n      <td>Xiaoqin Guo (Sun Yat-sen University, 510080, G...</td>\n      <td>Abstract\\n\\n23 BACKGROUND 24 The ongoing world...</td>\n      <td>\\n\\nCC-BY-ND 4.0 International license It is m...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>bbd9d63dc2c733c763770f62205ef9adeceb0127</td>\n      <td>Effects of temperature variation and humidity ...</td>\n      <td>Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...</td>\n      <td>Yueling Ma (Lanzhou University, 730000, Lanzho...</td>\n      <td>Abstract\\n\\nObject Meteorological parameters a...</td>\n      <td>Introduction\\n\\nIn December 2019, a novel coro...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"df['abstract_word_count'] = df['abstract'].apply(lambda x: len(x.strip().split()))\ndf['body_word_count'] = df['text'].apply(lambda x: len(x.strip().split()))\ndf.head()","execution_count":8,"outputs":[{"output_type":"execute_result","execution_count":8,"data":{"text/plain":"                                   paper_id  \\\n0  4602afcb8d95ebd9da583124384fd74299d20f5b   \n1  90b5ecf991032f3918ad43b252e17d1171b4ea63   \n2  c8a9e49416051620631005def6ea86c91cf86bc9   \n3  d3c2e2839498c613ee95739dce7052109750362c   \n4  bbd9d63dc2c733c763770f62205ef9adeceb0127   \n\n                                               title  \\\n0  SPINT2 inhibits proteases involved in activati...   \n1  The role of absolute humidity on transmission ...   \n2  FEP-based screening prompts drug repositioning...   \n3  Long-Term Persistence of IgG Antibodies in SAR...   \n4  Effects of temperature variation and humidity ...   \n\n                                             authors  \\\n0  Marco R Straus, Jonathan T Kinder, Michal Sega...   \n1  Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...   \n2  Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...   \n3  Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...   \n4  Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...   \n\n                                        affiliations  \\\n0  Marco R Straus, Jonathan T Kinder (University ...   \n1  Wei Luo (Boston Children's Hospital, 02215, Bo...   \n2  Zhe Li (Sun Yat-Sen University, 510006, Guangz...   \n3  Xiaoqin Guo (Sun Yat-sen University, 510080, G...   \n4  Yueling Ma (Lanzhou University, 730000, Lanzho...   \n\n                                            abstract  \\\n0  Abstract\\n\\nViruses possessing class I fusion ...   \n1                                                      \n2  Abstract\\n\\nThe new coronavirus COVID-19, also...   \n3  Abstract\\n\\n23 BACKGROUND 24 The ongoing world...   \n4  Abstract\\n\\nObject Meteorological parameters a...   \n\n                                                text  abstract_word_count  \\\n0  Introduction 9\\n\\nInfluenza-like illnesses (IL...                   76   \n1  Introduction\\n\\nSince December 2019, an increa...                    0   \n2  Introduction\\n\\nThe novel coronavirus 2019-nCo...                  172   \n3  \\n\\nCC-BY-ND 4.0 International license It is m...                  302   \n4  Introduction\\n\\nIn December 2019, a novel coro...                  145   \n\n   body_word_count  \n0             6116  \n1             1511  \n2             4749  \n3             1004  \n4             2437  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>paper_id</th>\n      <th>title</th>\n      <th>authors</th>\n      <th>affiliations</th>\n      <th>abstract</th>\n      <th>text</th>\n      <th>abstract_word_count</th>\n      <th>body_word_count</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>4602afcb8d95ebd9da583124384fd74299d20f5b</td>\n      <td>SPINT2 inhibits proteases involved in activati...</td>\n      <td>Marco R Straus, Jonathan T Kinder, Michal Sega...</td>\n      <td>Marco R Straus, Jonathan T Kinder (University ...</td>\n      <td>Abstract\\n\\nViruses possessing class I fusion ...</td>\n      <td>Introduction 9\\n\\nInfluenza-like illnesses (IL...</td>\n      <td>76</td>\n      <td>6116</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>90b5ecf991032f3918ad43b252e17d1171b4ea63</td>\n      <td>The role of absolute humidity on transmission ...</td>\n      <td>Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...</td>\n      <td>Wei Luo (Boston Children's Hospital, 02215, Bo...</td>\n      <td></td>\n      <td>Introduction\\n\\nSince December 2019, an increa...</td>\n      <td>0</td>\n      <td>1511</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>c8a9e49416051620631005def6ea86c91cf86bc9</td>\n      <td>FEP-based screening prompts drug repositioning...</td>\n      <td>Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...</td>\n      <td>Zhe Li (Sun Yat-Sen University, 510006, Guangz...</td>\n      <td>Abstract\\n\\nThe new coronavirus COVID-19, also...</td>\n      <td>Introduction\\n\\nThe novel coronavirus 2019-nCo...</td>\n      <td>172</td>\n      <td>4749</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>d3c2e2839498c613ee95739dce7052109750362c</td>\n      <td>Long-Term Persistence of IgG Antibodies in SAR...</td>\n      <td>Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...</td>\n      <td>Xiaoqin Guo (Sun Yat-sen University, 510080, G...</td>\n      <td>Abstract\\n\\n23 BACKGROUND 24 The ongoing world...</td>\n      <td>\\n\\nCC-BY-ND 4.0 International license It is m...</td>\n      <td>302</td>\n      <td>1004</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>bbd9d63dc2c733c763770f62205ef9adeceb0127</td>\n      <td>Effects of temperature variation and humidity ...</td>\n      <td>Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...</td>\n      <td>Yueling Ma (Lanzhou University, 730000, Lanzho...</td>\n      <td>Abstract\\n\\nObject Meteorological parameters a...</td>\n      <td>Introduction\\n\\nIn December 2019, a novel coro...</td>\n      <td>145</td>\n      <td>2437</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"***Drop dublicates data from text and abstract data***"},{"metadata":{"trusted":true},"cell_type":"code","source":"df.drop_duplicates(['abstract', 'text'], inplace=True)","execution_count":9,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"df.head(5)","execution_count":10,"outputs":[{"output_type":"execute_result","execution_count":10,"data":{"text/plain":"                                   paper_id  \\\n0  4602afcb8d95ebd9da583124384fd74299d20f5b   \n1  90b5ecf991032f3918ad43b252e17d1171b4ea63   \n2  c8a9e49416051620631005def6ea86c91cf86bc9   \n3  d3c2e2839498c613ee95739dce7052109750362c   \n4  bbd9d63dc2c733c763770f62205ef9adeceb0127   \n\n                                               title  \\\n0  SPINT2 inhibits proteases involved in activati...   \n1  The role of absolute humidity on transmission ...   \n2  FEP-based screening prompts drug repositioning...   \n3  Long-Term Persistence of IgG Antibodies in SAR...   \n4  Effects of temperature variation and humidity ...   \n\n                                             authors  \\\n0  Marco R Straus, Jonathan T Kinder, Michal Sega...   \n1  Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...   \n2  Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...   \n3  Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...   \n4  Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...   \n\n                                        affiliations  \\\n0  Marco R Straus, Jonathan T Kinder (University ...   \n1  Wei Luo (Boston Children's Hospital, 02215, Bo...   \n2  Zhe Li (Sun Yat-Sen University, 510006, Guangz...   \n3  Xiaoqin Guo (Sun Yat-sen University, 510080, G...   \n4  Yueling Ma (Lanzhou University, 730000, Lanzho...   \n\n                                            abstract  \\\n0  Abstract\\n\\nViruses possessing class I fusion ...   \n1                                                      \n2  Abstract\\n\\nThe new coronavirus COVID-19, also...   \n3  Abstract\\n\\n23 BACKGROUND 24 The ongoing world...   \n4  Abstract\\n\\nObject Meteorological parameters a...   \n\n                                                text  abstract_word_count  \\\n0  Introduction 9\\n\\nInfluenza-like illnesses (IL...                   76   \n1  Introduction\\n\\nSince December 2019, an increa...                    0   \n2  Introduction\\n\\nThe novel coronavirus 2019-nCo...                  172   \n3  \\n\\nCC-BY-ND 4.0 International license It is m...                  302   \n4  Introduction\\n\\nIn December 2019, a novel coro...                  145   \n\n   body_word_count  \n0             6116  \n1             1511  \n2             4749  \n3             1004  \n4             2437  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>paper_id</th>\n      <th>title</th>\n      <th>authors</th>\n      <th>affiliations</th>\n      <th>abstract</th>\n      <th>text</th>\n      <th>abstract_word_count</th>\n      <th>body_word_count</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>4602afcb8d95ebd9da583124384fd74299d20f5b</td>\n      <td>SPINT2 inhibits proteases involved in activati...</td>\n      <td>Marco R Straus, Jonathan T Kinder, Michal Sega...</td>\n      <td>Marco R Straus, Jonathan T Kinder (University ...</td>\n      <td>Abstract\\n\\nViruses possessing class I fusion ...</td>\n      <td>Introduction 9\\n\\nInfluenza-like illnesses (IL...</td>\n      <td>76</td>\n      <td>6116</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>90b5ecf991032f3918ad43b252e17d1171b4ea63</td>\n      <td>The role of absolute humidity on transmission ...</td>\n      <td>Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...</td>\n      <td>Wei Luo (Boston Children's Hospital, 02215, Bo...</td>\n      <td></td>\n      <td>Introduction\\n\\nSince December 2019, an increa...</td>\n      <td>0</td>\n      <td>1511</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>c8a9e49416051620631005def6ea86c91cf86bc9</td>\n      <td>FEP-based screening prompts drug repositioning...</td>\n      <td>Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...</td>\n      <td>Zhe Li (Sun Yat-Sen University, 510006, Guangz...</td>\n      <td>Abstract\\n\\nThe new coronavirus COVID-19, also...</td>\n      <td>Introduction\\n\\nThe novel coronavirus 2019-nCo...</td>\n      <td>172</td>\n      <td>4749</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>d3c2e2839498c613ee95739dce7052109750362c</td>\n      <td>Long-Term Persistence of IgG Antibodies in SAR...</td>\n      <td>Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...</td>\n      <td>Xiaoqin Guo (Sun Yat-sen University, 510080, G...</td>\n      <td>Abstract\\n\\n23 BACKGROUND 24 The ongoing world...</td>\n      <td>\\n\\nCC-BY-ND 4.0 International license It is m...</td>\n      <td>302</td>\n      <td>1004</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>bbd9d63dc2c733c763770f62205ef9adeceb0127</td>\n      <td>Effects of temperature variation and humidity ...</td>\n      <td>Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...</td>\n      <td>Yueling Ma (Lanzhou University, 730000, Lanzho...</td>\n      <td>Abstract\\n\\nObject Meteorological parameters a...</td>\n      <td>Introduction\\n\\nIn December 2019, a novel coro...</td>\n      <td>145</td>\n      <td>2437</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"df.drop_duplicates(['title','abstract'], inplace=True)\ndf.shape","execution_count":11,"outputs":[{"output_type":"execute_result","execution_count":11,"data":{"text/plain":"(1322, 8)"},"metadata":{}}]},{"metadata":{},"cell_type":"markdown","source":"***Preprocessing***\n"},{"metadata":{"trusted":true},"cell_type":"code","source":"#df = df.head(10000)","execution_count":null,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"import re\n\ndf['text'] = df['text'].apply(lambda x: re.sub('[^a-zA-z0-9\\s]','',x))\ndf['abstract'] = df['abstract'].apply(lambda x: re.sub('[^a-zA-z0-9\\s]','',x))\ndf['text'] = df['text'].apply(lambda x: re.sub('\\n\\n',' ',x))\ndf['abstract'] = df['abstract'].apply(lambda x: re.sub('\\n\\n',' ',x))\ndf['text'] = df['text'].apply(lambda x: re.sub('\\d+', '',x))\ndf['abstract'] = df['abstract'].apply(lambda x: re.sub('\\d+', '',x))\n","execution_count":12,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"\ndf['text'] = df['text'].apply(lambda x: x.lower())\ndf['abstract'] = df['abstract'].apply(lambda x: x.lower())\ndf.head(5)","execution_count":13,"outputs":[{"output_type":"execute_result","execution_count":13,"data":{"text/plain":"                                   paper_id  \\\n0  4602afcb8d95ebd9da583124384fd74299d20f5b   \n1  90b5ecf991032f3918ad43b252e17d1171b4ea63   \n2  c8a9e49416051620631005def6ea86c91cf86bc9   \n3  d3c2e2839498c613ee95739dce7052109750362c   \n4  bbd9d63dc2c733c763770f62205ef9adeceb0127   \n\n                                               title  \\\n0  SPINT2 inhibits proteases involved in activati...   \n1  The role of absolute humidity on transmission ...   \n2  FEP-based screening prompts drug repositioning...   \n3  Long-Term Persistence of IgG Antibodies in SAR...   \n4  Effects of temperature variation and humidity ...   \n\n                                             authors  \\\n0  Marco R Straus, Jonathan T Kinder, Michal Sega...   \n1  Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...   \n2  Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...   \n3  Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...   \n4  Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...   \n\n                                        affiliations  \\\n0  Marco R Straus, Jonathan T Kinder (University ...   \n1  Wei Luo (Boston Children's Hospital, 02215, Bo...   \n2  Zhe Li (Sun Yat-Sen University, 510006, Guangz...   \n3  Xiaoqin Guo (Sun Yat-sen University, 510080, G...   \n4  Yueling Ma (Lanzhou University, 730000, Lanzho...   \n\n                                            abstract  \\\n0  abstract viruses possessing class i fusion pro...   \n1                                                      \n2  abstract the new coronavirus covid also known ...   \n3  abstract  background  the ongoing worldwide ou...   \n4  abstract object meteorological parameters are ...   \n\n                                                text  abstract_word_count  \\\n0  introduction  influenzalike illnesses ilis rep...                   76   \n1  introduction since december  an increasing num...                    0   \n2  introduction the novel coronavirus ncov also k...                  172   \n3   ccbynd  international license it is made avai...                  302   \n4  introduction in december  a novel coronavirus ...                  145   \n\n   body_word_count  \n0             6116  \n1             1511  \n2             4749  \n3             1004  \n4             2437  ","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>paper_id</th>\n      <th>title</th>\n      <th>authors</th>\n      <th>affiliations</th>\n      <th>abstract</th>\n      <th>text</th>\n      <th>abstract_word_count</th>\n      <th>body_word_count</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>4602afcb8d95ebd9da583124384fd74299d20f5b</td>\n      <td>SPINT2 inhibits proteases involved in activati...</td>\n      <td>Marco R Straus, Jonathan T Kinder, Michal Sega...</td>\n      <td>Marco R Straus, Jonathan T Kinder (University ...</td>\n      <td>abstract viruses possessing class i fusion pro...</td>\n      <td>introduction  influenzalike illnesses ilis rep...</td>\n      <td>76</td>\n      <td>6116</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>90b5ecf991032f3918ad43b252e17d1171b4ea63</td>\n      <td>The role of absolute humidity on transmission ...</td>\n      <td>Wei Luo, Maimuna S Majumder, Diambo Liu, Canel...</td>\n      <td>Wei Luo (Boston Children's Hospital, 02215, Bo...</td>\n      <td></td>\n      <td>introduction since december  an increasing num...</td>\n      <td>0</td>\n      <td>1511</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>c8a9e49416051620631005def6ea86c91cf86bc9</td>\n      <td>FEP-based screening prompts drug repositioning...</td>\n      <td>Zhe Li, 1# , Xin Li, Yi-You Huang, Yaoxing Wu,...</td>\n      <td>Zhe Li (Sun Yat-Sen University, 510006, Guangz...</td>\n      <td>abstract the new coronavirus covid also known ...</td>\n      <td>introduction the novel coronavirus ncov also k...</td>\n      <td>172</td>\n      <td>4749</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>d3c2e2839498c613ee95739dce7052109750362c</td>\n      <td>Long-Term Persistence of IgG Antibodies in SAR...</td>\n      <td>Xiaoqin Guo, Zhongmin Guo, Chaohui Duan, Zelia...</td>\n      <td>Xiaoqin Guo (Sun Yat-sen University, 510080, G...</td>\n      <td>abstract  background  the ongoing worldwide ou...</td>\n      <td>ccbynd  international license it is made avai...</td>\n      <td>302</td>\n      <td>1004</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>bbd9d63dc2c733c763770f62205ef9adeceb0127</td>\n      <td>Effects of temperature variation and humidity ...</td>\n      <td>Yueling Ma, Yadong Zhao, Jiangtao Liu, Xiaotao...</td>\n      <td>Yueling Ma (Lanzhou University, 730000, Lanzho...</td>\n      <td>abstract object meteorological parameters are ...</td>\n      <td>introduction in december  a novel coronavirus ...</td>\n      <td>145</td>\n      <td>2437</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"text = df.drop([\"paper_id\", \"abstract\", \"abstract_word_count\", \"body_word_count\", \"authors\", \"title\", \"affiliations\"], axis=1)","execution_count":14,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"text.head(5)","execution_count":15,"outputs":[{"output_type":"execute_result","execution_count":15,"data":{"text/plain":"                                                text\n0  introduction  influenzalike illnesses ilis rep...\n1  introduction since december  an increasing num...\n2  introduction the novel coronavirus ncov also k...\n3   ccbynd  international license it is made avai...\n4  introduction in december  a novel coronavirus ...","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>text</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>introduction  influenzalike illnesses ilis rep...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>introduction since december  an increasing num...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>introduction the novel coronavirus ncov also k...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>ccbynd  international license it is made avai...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>introduction in december  a novel coronavirus ...</td>\n    </tr>\n  </tbody>\n</table>\n</div>"},"metadata":{}}]},{"metadata":{"trusted":true},"cell_type":"code","source":"docs = []\nfor x in range(0,len(text)):\n    docs.append(str(text.iloc[x]['text']))","execution_count":16,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"print(docs[5])","execution_count":17,"outputs":[{"output_type":"stream","text":" arrays of regularly spaced nucleosomes dominate chromatin and are often phased by alignment to reference sites like active promoters how the distances between nucleosomes spacing and between phasing sites and nucleosomes are determined remains unclear and specifically how atp dependent chromatin remodelers impact these features here we used genomewide reconstitution to probe how saccharomyces cerevisiae atp dependent remodelers generate phased arrays of regularly spaced nucleosomes we find that remodelers bear a functional element named the ruler that determines spacing and phasing in a remodelerspecific way we use structurebased mutagenesis to identify and tune the ruler element residing in the nhp and arp modules of the ino remodeler complex generally we propose that a remodeler ruler regulates nucleosome sliding direction bias in response to epigenetic information this finally conceptualizes how remodelermediated nucleosome dynamics determine stable steadystate nucleosome positioning relative to other nucleosomes dna bound factors dna ends and dna sequence elementsnuclear dna is packaged into chromatin based on a repeating building block the nucleosome core particle ncp kornberg  olins and olins   where  base pairs bp of dna are wound around a histone protein octamer kornberg and lorch  luger et al  olins and olins   packaging by nucleosomes orchestrates all genomic processes lai and pugh   nucleosomes mainly occur in regular arrays where they are aligned to each other such that the lengths of linker dna between ncps are about constant within an array linker lengths may vary among arrays in the same cell baldi et al b chereji et al  ocampo et al  valouev et al  and differ on average between cell types and species van holde   arrays are often phased ie aligned relative to a genomic reference point a combination of both in vivo studies ganapathi et al  hartley and madhani  kubik et al  tsankov et al  van bakel et al  yan et al  yarragudi et al  and in vitro reconstitutions indicated that these  senior author  these authors contributed equally  lead contact  correspondence pkorberlmude hopfnergenzentrumlmude sebastianeustermannemblde genomic alignment points or barriers often reflect the binding of abundant sequencespecific dna binding proteins like the general regulatory factor grfs reb abf or rap in budding yeast or other architectural factors like ctcf in mammals wiechens et al  or phaser in flies baldi et al a  throughout eukaryotes phased arrays are prominent at active promoters nucleosomedepleted regions ndrs at the core promoter are flanked by arrays that begin with the so called  nucleosome close to the transcription start site tss and cover the gene body baldi et al  lai and pugh   this organization is important for transcription fidelity as mutants with impaired array phasing show aberrant transcription initiation challal et al  hennig et al  kubik et al  pointner et al  smolle et al   while nucleosome arrays are likely the most pervasive and longest known chromatin organization their generation is still not explained specifically regular spacing requires fixed distances between nucleosomes and phasing requires a fixed distance between array and reference point what sets these distances in vivo and in vitro data suggest that atp dependent chromatin remodeling enzymes remodelers are key to the answer remodelers are conserved in eukaryotes and mobilize reconfigure or disassemblereassemble nucleosomes upon atp hydrolysis clapier and cairns  clapier et al   they are subdivided into the swisnf iswi chd and ino families according to their main atpase sequence features besides the core atpase remodelers often contain additional domains and subunits that bind the nucleosome regulate activity and targeting and convert their dna tracking activity into the remodelerspecific chemomechanical reaction for example nucleosome disassembly is accomplished only by swisnf family members and histone exchange only by ino family members while nucleosome sliding is catalyzed by most remodelers particularly relevant for array generation is an atpdependent nucleosome spacing activity by which some remodelers convert irregular arrays into arrays of regularly spaced nucleosomes remodelers of the iswi chd and ino ito et al  tsukiyama et al  udugama et al  vargaweisz et al   but not of the swisnf family show spacing activity this activity was suggested to rely on a lengthsensor mechanism yang et al  zhou et al  where nucleosome sliding rate is regulated by linker dna length sliding one nucleosome back and forth between two other nucleosomes with a linker lengthdependent velocity would center a nucleosome at steady state when both flanking linkers have the same length while the lengthsensor mechanism may equalize linker lengths and thereby generate spacing distance regularity it does not by itself determine spacing distance length in absolute terms this would reciprocally depend on nucleosome density however spacing in vivo gossett and lieb  hennig et al  van bakel et al   as well as generated in vitro lieleg et al  zhang et al   remained constant despite changes in nucleosome density this was called active packing zhang et al  or clamping lieleg et al   but it remained unclear if remodeler or nucleosome features led to such densityindependent spacing structural studies suggested that the yeast iswa remodeler contacts a neighboring nucleosome and may set the linker length by a protein ruler yamada et al   two iswi family remodelers yeast iswa and isw each generated regular arrays aligned at dnabound reb or abf in vitro but with different spacing at the same nucleosome density  this points towards a remodelerspecific linker length determining ruler mechanism also suggestive of a builtin ruler ino required a minimum linker length for nucleosome sliding zhou et al  and recognized linker dna via a structural module that was important for sliding  the ruler metaphor may indeed describe a remodeler mechanism that measures and sets the phasing and spacing distances of arrays however so far it is mainly suggestive and has to be substantiated in molecular terms this would be exceedingly convoluted in vivo but requires a defined system that allows to assay the generation of phased regular arrays by remodelers and to dissect if and how a ruler mechanism is at work are there rulers within some or all remodelers with spacing activity are linker length vs distance to barrier determined in the same or different way are rulers autonomous or does the outcome depend on nucleosome density or underlying dna sequence ultimately is it possible to tune a ruler ie can a remodeler be mutated to generate arrays with altered spacing andor phasing distances here we used genomewide in vitro chromatin reconstitution with purified remodelers   accompanying paper oberbeckmann  krietenstein et al to answer these questions all yeast remodelers with spacing activity iswa isw chd and ino have rulers that are largely autonomous regarding underlying dna sequence but some may respond to nucleosome density remodelerspecific rulers mechanistically explain earlier in vivo observations structureguided mutations in recombinant ino complexes led to shorter or longer spacing and phasing distances and showed that these quantities may be uncoupled finally we propose a model how remodeler rulers position nucleosomes by regulating sliding direction bias according to epigenetic information in the nucleosome environment purified yeast remodelers figure sa and the barrier reb or the restriction enzyme bamhi which generates double strand breaks dsbs that also amount to nucleosome positioning barriers accompanying paper oberbeckmann  krietenstein et al  resulting nucleosome patterns were analyzed by mnaseseq the effective histonetodna mass ratio during sgd was varied from  to  yielding low medium and high nucleosome densities reflected in increasingly extensive mnaseladders at the same mnase digestion conditions  figure b  nucleosome density variation was instrumental to distinguish if linker lengths and phasing distances depended on nucleosome density andor remodeler featuresino isw iswa and chd but not fun align regular arrays at the barrier reb we tested all yeast remodelers with known spacing activity ino isw iswa and chd lusser et al  stockdale et al  torigoe et al  tsukiyama et al  udugama et al  as well as the fun remodeler for which it was unclear if it has spacing activity awad et al   ino isw iswa and chd each in combination with reb generated phased regular arrays at promoters with reb sites red shaded top of heat maps in figure c  while fun did not  figure sb  this clarifies that fun does not have regular array generation and alignment activity previously chd purified from budding yeast did not show much effect in genomewide reconstitutions  this was maybe due to fulllength chd tending to aggregate in vitro which is why truncated chd constructs were often used patel et al   here we leveraged our finding that recombinant fulllength chd is stabilized in complex with recombinant fact complex farnung et al  and achieved in vitro array generation and alignment also by chd the heat map patterns  figure c  and even more the corresponding composite plots for the rebbound genes only  figure d  suggested that the distance of arrays to the barrier reb as well as the linker lengths varied with nucleosome density in a remodelerspecific way for all remodelers with spacing activity array extent increased with growing density consistent with greater nucleosome availability and processive spacing activity array extent at high density was larger than in our previous reconstitutions  ie we achieved higher densities here adding more remodeler after half of the incubation time did not change the array distances of resulting patterns confirming nonlimiting remodeling activity and steady state conditions  figure sc  remodelers set phasing and spacing distances symmetrically around barriers to better assess distances to barrier phasing and linker lengths spacing we aligned the mnaseseq data for each remodelerbarrierdensity combination to either in vivo reb sites or bamhi sites  figure a  for each replicate  figure sa c we called nucleosome peaks and determined the distances to barrier and linker lengths as defined in figure b  all remodelers symmetrically aligned regular arrays to bamhi sites which are palindromic and therefore inherently symmetrical and most of them also to reb sites figures a sa b regardless of site orientation and position relative to genes groups  to  figure sa b however if ino aligned arrays at promoter reb sites groups  to  figure  sa  accompanying paper by oberbeckmann  krietenstein et al  nucleosome occupancy peak height was higher over genic versus nongenic regions at low and medium nucleosome density leading to asymmetric patterns with regard to peak heights in groups  and  reb site orientation had no effect group  vs  this asymmetry in nucleosome occupancies reflected that positioning of  nucleosomes per definition the first nucleosomes downstream of transcription start sites ie at gene starts was not only guided by reb bound to promoter sites but also synergistically by underlying dna shape features accompanying paper oberbeckmann  krietenstein et al  we recapitulated here that ino was able to position in vivolike  nucleosomes in the absence of a barrier at low and medium densities  figure sc d this synergism between reband dna shapeguided  positioning at low and medium density resulted in higher occupancy at the  nucleosomes which are alignment points for  nucleosomes and so on therefore all array peaks over genes were higher than their counterparts over nongenic regions however such synergism was not seen at high density where in vivolike  nucleosomes positioning by ino alone was much less pronounced  figure sc d this inability was not due to a general inability of ino to slide densely packed nucleosomes as ino could generate rebaligned arrays at these high nucleosome densities too figures c d a sa b  nonetheless this activity was apparently incompatible with or dominant over dna shapeguided nucleosome positioning see discussion this showed again that our here generated high nucleosome density was higher than the nucleosome density used previously  otherwise in vivolike  nucleosome positioning by ino would not have been clearly observed in our earlier study heat maps of mnaseseq data for sgd chromatin of the indicated nucleosome density and after incubation with indicated remodelers and reb chd refers to the chdfact complex heat maps are aligned at in vivo  nucleosome positions and sorted according to decreasing top to bottom antireb slimchip score shown in leftmost heat map horizontal red shading highlights genes with strong in vivo rebbinding in their promoters merged data of replicates are shown individual replicates in figure sb c d composite plots for mnaseseq data averaged over the indicated number of replicates n as in panel b but only for genes highlighted in red in panel bin this context we also tested if fun positions in vivolike  nucleosomes on its own but it did not  figure sd  in contrast to nucleosome peak heights nucleosome peak positions and therefore corresponding phasing and spacing distances were not significantly affected across groups  to  for all remodelers including ino  figure sb  therefore all remodelers symmetrically generated phasing and spacing distances at reb and bamhi sites which warranted averaging over the upand downstream values resulting values were plotted in different ways to the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint facilitate multidimensional comparisons  figure  c e as all remodelers generated linker lengths independently of the barrier type we combined linker length values for both barriers reb and bamhi figure c  linker length determination relied on nucleosome peak calling which was often not possible beyond the  nucleosomes at low nucleosome density  figure sa  so that linker length data for low density conditions were more sparse even absent for iswaremodelerspecific rulers set spacing in a densityindependent or dependent way to compare spacing generated by different remodelers at different nucleosome densities we focused on the averaged length of linker   figure b  which was most accessible across all nucleosome densities chd generated the shortest  bp and iswa a bit longer  bp linker  lengths without significant effects by nucleosome density  figure  d e isw generated rather constant spacing  bp at low and medium but tighter spacing  bp at high density for ino linker lengths steadily increased with decreasing density from  to  bp we concluded that linker lengths and their dependencies on nucleosome density were remodelerspecific and interpreted this as follows spacing activity of a remodeler has two aspects on the one hand the remodeler equalizes linker lengths leading to regularity in arrays which is the classical definition of spacing activity ito et al  vargaweisz et al   on the other hand the resulting linkers have a certain length in our purified system this may either be determined by nucleosome density andor by a remodelerintrinsic feature following yamada et al   we call a remodeler feature that sets nucleosome spacing a ruler we use this term also for the feature that sets the distance to barriers see below indicative for a remodeler ruler is remodelerspecific clamping ie if constant spacing is generated at different nucleosome densities  clamping and different remodelers generate different spacing  remodelerspecific which shows that spacing depends on remodelerintrinsic and not nucleosomeintrinsic properties lieleg et al   we saw remodelerspecific clamping for chd at all for iswa at high versus medium and for isw at medium versus low densities  figure c e as none of the remodelers with spacing activity can disassemble nucleosomes clapier and cairns  and thereby affect nucleosome density their rulers can only set their respective linker lengths if these are shorter than or equal to the densitydetermined linker length at equidistant nucleosome distribution accordingly chd and iswa set their rulerspecified linker lengths at all and isw at medium and low densities isw had to generate shorter linkers at high density and ino either did not have a ruler or the ruler responded to changes in nucleosome density in vitro mononucleosome assays suggested that ino requires at least  bp of nucleosomefree dna for nucleosome sliding zhou et al   while it generated  bp linkers in trinucleosomes udugama et al   here at high nucleosome density ino generated linkers of about  bp consistent with previous observations we tried to enforce even tighter spacing by increasing nucleosome density this did not decrease spacing and phasing distances but peak heights  figure  sb c probably due to increased aggregation without effective increase in nucleosome density of soluble chromatinremodeler type barrier type and nucleosome density determine distance to barrier the findings for the distance to barrier were more complex than for lengths of linker   figure c e first the distance to barrier depended on the barrier type  figure c  it was always longer for reb than for bamhi generated dna ends with the largest difference for iswa and the smallest for chd the dna footprint size of s cerevisiae reb is not known possibly  bp as for the s pombe reb dna binding domain jaiswal et al   this would contribute  bp to the distance to barrier  figure b  and could explain the differences between distance to reb vs bamhi sites for chd but not for the other remodelers therefore ino isw and iswa but not chd aligned nucleosomes differently at reb versus at dsbs second the distance to dna ends was mostly similar to linker lengths for ino isw and iswa arguing that these remodelers but not chd used a dna end in a similar way as a neighboring nucleosome for nucleosome alignment third distances to barriers depended on nucleosome density in a similar way as linker lengths for all remodelers but ino where distances to both barriers varied less between low and medium density than linker length we concluded that there are remodelerspecific differences in how a nucleosome is positioned next to another nucleosome versus next to a barrier like reb versus next to a dna end and how this depends on nucleosome density this is again a clear case of different remodelers generating different nucleosome positioning although starting from the same sgd chromatin which argues for remodelerspecific rulers governing nucleosome positioning d which we interpreted as processive spacing activity along the arrays as long as nucleosomes were sufficiently provided at low density isw chd and especially ino still generated high  nucleosome peaks  figure sa  in contrast to iswa for which these peaks were less pronounced and  nucleosome peaks could not be discerned we suggest that iswa is less processive than other remodelers in bringing nucleosomes next to barriers at low densities distance to barrier site  bp normalized dyad density remodelers generate similar arrays on all but more effectively on eukaryotic dna sequences the same linker lengths in arrays at bamhi and reb sites  figure c  at reb sites in groups  to  and the symmetry of nucleosome distances to reb sites in groups  to   figure sa b suggested that remodeler rulers position nucleosomes independently of dna sequence flanking the barriers nonetheless there are evolved dna features at promoters especially for ino accompanying paper oberbeckmann  krietenstein et al  that affected occupancies peak heights not positions figure sa  which may also be true for evolved nucleosomefavoring dinucleotide periodicities satchwell et al  in gene bodiesto rigorously disentangle these contributions we tested the remodelerbarrierdensity combinations also with sgd chromatin of s pombe and e coli genomic plasmid libraries  figures a a b sa including the steady state control  figure sb  we did not observe substantial differences in spacingphasing distances on these genomes for all remodelers but some replicates especially at medium and low density showed lower relative occupancies for the e coli genome we concluded that all remodelers align arrays at reb or dsbs regardless of the underlying sequence nonetheless they are more effective in terms of relative occupancies on eukaryotic genomes likely due to dinucleotide periodicities zhang et al  authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint ino complexes mutated in the arp andor nhp module it was unexpected that the clamping criterion did not clearly show a ruler for ino  figure c e because the ino structure suggested modules that bind extranucleosomal dna and could serve as ruler  to clarify we took advantage of the biochemical accessibility of our recombinant ino preparation the modular authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint ino composition and the highresolution structures to generate candidate mutations that may tune and thereby reveal inos ruler the ino complex has two modules with a likely role in ruler function first the arp module consisting of nactin arp arp taf and ies  figure a  it binds the ino main atpase hsa domain which is structured as a long helix with a kink that subdivides it into the hsaa and hsaa part  both bind to extranucleosomal dna and mutating dna contacting lysine residues in hsaa or hsaa to glutamines hq and hq mutant respectively figure b cd impaired and combining both mutations hq mutant abolished mononucleosome centering activity the second nhp module binds the ino atpase nterminus and contains the hmg box nhp subunit along with ies ies and ies  figure a e this module is speciesspecific and affects the processivity and extranucleosomal dna requirements in mononucleosome sliding assays zhou et al   c authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint calculating a homology model for nhp based on another hmg box protein tfam ngo et al   we inferred and mutated amino acid residues putatively involved in nhpdna interactions hmgii mutant figure f g these mutations were also combined with the hq or hq mutants hmgiihq and hmgiihq further we prepared recombinant ino complex without any nhp module subunits nhp mutant no truncation of the ino atpase nterminus or a version where the ino atpase lacked residues  ino n mutant which removes the assembly platform for the nhp module  figure a  authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the ino mutant complexes reveal a multilayered ruler all mutant complexes were assayed like the wild type wt ino complex  figures ae a d sa sab wt ino was assayed again alongside with matching sgd chromatin comparing these replicates  figure c  with previous values for wt ino  figure d  reflected variability in preparing sgd chromatin but at the same time the robustness of the overall effects all tested ino mutants generated steadystate patterns  figure sb  and differed from wt ino in forming aligned arrays in the following ways first all mutants besides the hq mutant which was almost inactive figure sa  as expected  generated phased regular arrays but with varying effectiveness and altered distance to one or both barrier types andor linker lengths compared to wt ino  figures de d  this revealed that also ino has a ruler to which both the arp and the nhp module contribute second the hq showed stronger effects than the hq mutation  figure d  both increased the distances to both barriers while hq increased linker length at all densities hq gained clamping activity ie linker length hardly depended on nuclesome density both mutations uncoupled distance to dna ends from linker lengths in contrast to wt ino  figure de  only for hq linker  length depended on barrier type  figure b  we concluded that the arp module especially via hsaa helixdna interactions is threefold involved in spacing alignment to barrier and responding to nucleosome density third the nhp module subunits contributed to the ruler mainly through the hmg box of nhp as the respective point mutations hmgii mutant mimicked the effects upon lack of all nhp module subunits nhp mutant  figure c d with these mutations distances to both barriers were not much affected but linker length depended less on density ie clamping was gained similar to the hq mutation effects of the combined hmgiihq and hq mutations were dominated by the hq mutations but with reduced effects on distance to barriers  figure e  even though the nhp hmg box was a prime candidate for sensing extranucleosomal dna its contribution was minor compared to the hsa helix contribution fourth the ino n mutation affected the distance to reb and even more to dna ends but gained clamping less strongly than the hmgii or nhp mutations  figure d  the ino n mutant lacked the complete nhp module but also the ino atpase nterminus and taf  figure sa  which may account for the differential effectsfifth the ino n  hq and hq mutations most drastically affected distance to bamhi sites but in opposite ways  figures cd cd  effects on nucleosome stimulated atpase activity versus on ruler function are not strictly coupled the wt ino atpase activity is stimulated by nucleosomes and inhibited about twofold in the presence of reb accompanying paper oberbeckmann  krietenstein et al  this relative inhibition by reb was not seen or less pronounced for the mutated ino complexes  figure e  the atpase activity of hmgii and nhp mutants was similar to that of wt ino in the presence of reb the ino n mutant had intermediate activity we concluded that all tested mutants were affected both with regard to atpase activity and with regard to their ruler but that both effects were not strongly coupledchaetomium thermophilum ino core complex suggests speciesspecific ruler the ino core complex of c thermophilum which we previously used for cryoem studies  corresponds to the s cerevisiae ino n mutant as it also lacks its ino atpase nterminus it showed stronger clamping and generated shorter linkers and distances to reb than ino n at all densities and much shorter linkers and distances to both barriers than s cerevisiae wt ino at low and medium densities  figure bd  this suggests that inos ruler may be speciesspecific discussion our study answers one of the oldest questions in chromatin research what determines the spacing and phasing distances of nucleosome arrays in absolute terms the solution to this question are atp dependent remodelers from the iswi chd and ino families with spacing activity these do not only equalize linker lengths but as we reveal here bear rulers for setting distances between two adjacent nucleosomes and between nucleosomes and other alignment points remodeler rulers explain previous in vivo observations rulers combined with barriers mechanistically explain in vivo observations that involved iswa isw chd and ino in  nucleosome positioning andor array regularity and phasing hennig et al  kubik et al  ocampo et al  parnell et al  pointner et al  van bakel et al  whitehouse et al  yen et al   the average s cerevisiae linker length of  bp thomas and furber   results from combined contributions of iswa and chd ocampo et al  as we show that iswa and chd rulers authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint generate linkers of about  and  bp respectively the  bp average linker speaks for iswa contributing globally more than chd indeed lack of isw in vivo globally shortened linkers while lack of chd affected global spacing only mildly kubik et al  ocampo et al   locally high transcription rate correlates with shorter spacing ocampo et al   which points to increased chd contribution probably due to increased chd recruitment by elongating rna polymerase simic et al   remodelerspecific rulers explain how iswa isw and ino affect  nucleosome positioning in vivo kubik et al  parnell et al  whitehouse et al  yen et al  and in vitro  especially in combination with rsc rsc and swisnf are the only yeast remodelers that disassemble nucleosomes clapier and cairns  clapier et al   particularly at promoter ndrs brahma and henikoff  ganguli et al  hartley and madhani  kubik et al  kubik et al  parnell et al  rawal et al  van bakel et al   by definition a promoter ndr has low nucleosome density therefore remodeler rulers will set distances to ndrbound barriers as measured here at low or medium nucleosome density in vivo distances between reb and  nucleosomes are  bp  figure sb  rhee and pugh   which are within remodelerspecific distances to reb at medium or low density  bp for ino  bp for isw  bp for iswa isw and ino contribute more to  nucleosome positioning in vivo than iswa kubik et al  as their long rulers are more suited for setting long distances across ndrs conversely the short chdruler hardly contributes to  positioning in vivo kubik et al  ocampo et al  van bakel et al   these different ruler characteristics explain why iswa and chd are mainly involved in spacing nucleosomes into densely packed arrays and why isw and ino mainly use their ruler for  alignment at ndrs in vivo this resolves the conundrum why yeast has two remodelers ino and isw that seemingly generate too wide spacing compared to average in vivo spacing we do not preclude that other mechanisms like recruitment via histone modifications or transcription factors also affect where each remodeler is activefunctional and structural identification of remodeler rulers the protein ruler model was first proposed for iswa yamada et al   it suggested that iswa shortens the linker until its ruler contacts the neighboring nucleosome but did not conceptualize why this would lead to a stable nucleosome position we built on and expanded this model identified remodeler rulers via their functionality and pinpointed the ino ruler also in structural terms on the functional level a ruler is revealed if a the same remodeler generates the same phasing andor spacing distances although it works on chromatin with varying nucleosome density clamping activity or b different remodelersdifferent mutant versions of the same remodeler generate different phasing andor spacing distances although they all work on the same chromatin remodelerspecific phasingspacing for the ino complex we found that the nhp module especially the ino nterminus as well as the arp module especially the inohsahelix contributed to the ruler function lack of the ino nterminus concomitant with lacking the nhp module allowed ino eg to slide nucleosomes closer to dna ends maybe for steric reasons while impaired dna traction during remodeling due to compromised inohsa helixdna interactions had the opposite effect it remains to be elucidated how exactly such modules within the multisubunit organisation relay barrier information to the core atpaseremodeler rulers regulate nucleosome sliding direction bias in response to nucleosome environment we propose an overarching framework for this relay that amounts to a widely applicable remodeler ruler principle figure   a remodeler may slide a nucleosome either to the left or to the right from a given position if there is no bias for sliding in either direction the nucleosome will experience a random walk along the dna regions c in in three hypothetical examples figure  a  net nucleosome movement in one direction gangaraju and bartholomew  langst et al  mcknight et al  stockdale et al  udugama et al  yang et al  zhou et al  requires an overall sliding direction bias in this direction we conceptualize a remodeler ruler as a remodelerintrinsic feature that generates an overall sliding direction bias in response to the epigenetic information in the environment of the nucleosome that the remodeler is remodeling the bias may originate from differences eg in binding orientation atpase activity sliding rate or processivity and is regulated by interaction of the ruler with a generalized barrier this may be a grf a dsb a neighboring nucleosome or a dna sequence element histone modifications variants may modulate as well while the microscopic details may differ for different remodelers and information authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoi org org  input the overall regulation of sliding direction bias by the ruler will share three key elements that constitute the ruler mechanism first the ruler has a certain reach regions a  b in figure a  within which it interacts with the barrier second if the position from where the remodeler slides the nucleosome is within region b the interaction between ruler and barrier biases overall sliding direction towards the barrier red curve is above green curve eg due to binding energy gained upon orienting the remodeler towards vs away from the barrier third if the nucleosome is in region a the rulerbarrier interaction disfavors sliding towards relative to sliding away from the barrier green curve is above red curve eg because the ruler gets sterically in the way our study determined the length of region a for different remodeler and barrier types and conditions region b and exact curve shapes will have to be determined in future studies if these three key elements are met resulting fluxes lead to steadystate nucleosome placement at a defined position relative to the barrier stippled vertical arrows throughout figure   this position is a selfstabilizing dynamic equilibrium point intersection of red and green curves without sliding direction bias here but with biases towards this point from neighboring positions this model applies to how a remodeler with ruler stably positions a nucleosome next to a grf as well as to another nucleosome and therefore explains both spacing and phasing it also explains densityindependent clamping as long as a remodeler is processive enough to fortuitously bring nucleosomes into region b of a barrier also at low density the ruler mechanism will keep the nucleosome at the dynamic equilibrium point nonetheless the model can also accommodate sensing of nucleosome density and barrier type eg if the ruler offers a hierarchy of interaction points brahma et al   ino mutants showed not concerted but uncoupled effects on distances to reb dna ends and nucleosomes even if the same module like the nhp module was differentially mutated chd generated shorter linker lengths  bp than distances to dna ends or reb  bp for chd reb may be a hard barrier while nucleosomes are soft barriers as they are partially invaded by the ruler indeed chd partially unwraps nucleosomal dna farnung et al   the way how different remodeler rulers interact with different barriers requires clarification and we outline our model figure  in terms of extensionless point particles but actual footprints have to be taken into account the model is fully compatible with the ruler ie the dna binding domain dbd of chd or drosophila acf yang et al   introducing bias via sensing extranucleosomal dna length indeed differently long extranucleosomal dna in monoor oligonucleosome sliding assays amounts to different distances to barriers like dna ends or other nucleosomes our model is fully consistent with previous data and models but offers an alternative interpretation and is more widely applicable eg to stable nucleosome positioning at only one barrier and not only inbetween two barriers we introduced our model in terms of overall sliding direction bias more specifically the model may refer to differential regulation of sliding rates ie the yaxis in figure a could correspond to overall sliding rate to the left or to the right if sliding rates are reciprocally regulated example  figure a  the sum of absolute sliding rate values is constant at each position  figure b  but not upon asymmetric regulation of sliding direction examples  and  figure a  as special case example  figure a b the dynamic equilibrium point may correspond to a minimum of absolute sliding rate this case corresponds to the kinetic release model manelyte et al  rippe et al   which posits that remodelers position nucleosomes at sites where the nucleosome is the locally poorest substrate for remodelingrulerregulated sliding the unifying principle for nucleosome positioning by remodelers as nucleosome positions are defined by the dna sequence bound by the histone octamer all mechanisms that generate consistent nucleosome positions across many genome copies must select certain dna sequences in competition with other sequences as shown here and in the accompanying paper oberbeckmann  krietenstein et al  remodelers may mediate this selection in two ways on the one hand a remodeler may directly choose a sequence eg ino turns dna shape features into  nucleosome positions at promoters accompanying paper oberbeckmann  krietenstein et al on the other hand a remodeler ruler may place a nucleosome at a rulerdetermined distance to a barrier eg isw aligns nucleosomes to reb and generates a regular array by aligning a second nucleosome to the first and so on in the former case the resulting nucleosomal sequence is directly selected for its sequence features while in the latter case it is indirectly selected without regards for its sequence features but merely for its position relative to the barrier as we show here by using reb sites in s pombe and e coli genomes our ruler model unifies these positioning mechanisms the generalized barrier also encompasses dna sequence elements with which a remodeler ruler interacts such that sliding direction bias is regulated  figure c  this explains observations for hybrid chd remodelers where the chd dbd was replaced with heterologous sequencespecific dbds donovan et al  mcknight et al  mcknight et al   such hybrid chd remodelers slide nucleosomes faster towards the cognate site of the heterologous dbd if it was in reach of this site until the nucleosome became positioned on the site in our model the heterologous dbd is a remodeler ruler as a dna sequence element as barrier is no hindrance for nucleosome sliding the remodeler may slide the nucleosome onto this site this prevents ruler binding to the site abolishes the increase in sliding rate linked to ruler binding and makes a nucleosome on the cognate site a poorer nucleosome sliding substrate than at neighboring positions  figure c  right which corresponds to the kinetic release model as noted  our model now adds that sliding from neighboring positions will always within ruler reach convene at the cognate site and stabilize this position even if there is no local sliding rate minimum as long as the ruler regulates sliding direction bias according to the three key elements outlined above  figure c left in vivo there are many ways that may regulate nucleosome positioning by remodelers eg by recruitment by architectural factors by nucleosome density fluctuations or by histone variants and modifications possibly in the context of elongating polymerases nonetheless we expect that the regulation of nucleosome sliding direction bias via builtin sensing and processing of information in the nucleosome environment ie a remodeler ruler will be at the heart of each nucleosome positioning mechanism organisms as source for materials used in experiments the pgp yeast genomic plasmid library was expanded from the clonal plates provided by open biosystems for generation of genomic plasmid libraries the s pombe strain hu ekwall group and e coli strain atcc  strain  affymetrix were used ino wildtype and mutant complexes chd and fact were expressed in trichoplusiani insect cells spodoptera frugiperda sf insect cells were used for virus production the saccharomyces cerevisiae iswa and fun remodelers were purified from the correspondingly taptagged yeast strains ioctap funtap as provided by open biosystems yeast isw was purified from strain ytt iswxflag tsukiyama et al   reb was purified from e coli bl de cd cells the drosophila embryo histones were prepared from the drosophila melanogaster strain oregonr embryonic d melanogaster histones wholegenome plasmid libraries and salt gradient dialysis embryonic d melanogaster histone purification the preparation of embryonic d melanogaster histones octamers was carried out as described in krietenstein et al  and simon and felsenfeld   briefly  g of  hours old d melanogaster embryos were dechorionated in   sodium hypochlorite washed with dh and resuspended in  ml lysisbuffer  mm khepes ph   mm kcl  mm mgcl  mm edta  mm egta  mm dtt  mm pmsf   glycerol embryos were homogenized yamamoto homogenizer filtered through cloth and centrifuged at  g for  min nuclei brownish light pellet were washed  times with  ml sucrosebuffer  mm khepes ph   mm kcl  mm mgcl  mm edta  mm egta  mm dtt  mm pmsf   sucrose and resuspended in  ml sucrosebuffer containing  mm cacl to obtain mononucleosomes nuclei were incubated for  min at  c with  units mnase sigmaaldrich reaction was stopped with  mm edta nuclei were pelleted and resuspended in  ml te  mm trishcl ph   mm edta containing  mm dtt and  mm pmsf followed by  to  min of rotation at  c nuclei were centrifuged for  min at  g at  c solubilized mononucleosomes are found in the supernatant which was applied to a preequilibrated hydroxyapatite column after washing the hydroxyapatite column with  m kcl histone octamers were eluted with  m kcl concentrated and stored in   glycerol and x complete roche protease inhibitors without edta at  cwholegenome plasmid library expansion the s cerevisiae genomic plasmid library pgp was originally described in jones et al  and purchased as a clonal glycerol stock collection from open biosystems library expansion was carried out via a singer rotor plating machine singer instruments  rounds  replicas after  hours colonies were combined into x l of lb medium containing  gml kanamycin and grown for  hours cells were harvested and subjected to plasmid giga preparation pc   kit machereynagel for s pombe and e coli plasmid library generation genomic s pombe hu and e coli type b cells atcc  strain  affymetrix dna was fragmented by a limited sauiiia or alui digest fragmented dna was ligated into pjet vector thermofisher scientific and transformed into electrocompetent dh cells cells were plated on lb plates containing  gml ampicillin grown for   hours combined in lb medium containing  gml ampicillin and grown for another  hours plasmids was extracted with plasmid mega preparation kit pc  kit machereynagel  salt gradient dialysis sgd for low medium and high assembly degrees  g of plasmid library dna s cerevisiae s pombe or e coli was mixed with   or  g of drosophila embryo histone octamers respectively in  l assembly buffer  mm trishcl ph   m nacl  mm edta   igepal ca  g bsa samples were transferred to slidealyzer mini dialysis devices which were placed in a  l beaker containing  ml of high salt buffer  mm trishcl ph   m nacl  mm edta   igepal ca  mm mercaptoethanol and dialyzed against a total of  l low salt buffer  mm trishcl ph   mm nacl  mm edta   igepal ca  mm mercaptoethanol added continuously via a peristaltic pump over a time course of  h while stirring mercaptoethanol was added freshly to all buffers after complete transfer of low salt buffer samples were dialyzed against  l low salt buffer for  h at room temperature dna concentration of the sgd chromatin preparations was estimated with a ds spektrophotometer denovix and could be stored at  c for several weeks to estimate the extent of the assembly degree an aliquot of the sample was subjected to mnase digestion as described below for mnaseladder read out purifications of chromatin remodeling enzymes expression and purification of ino complex and respective mutants exact strategy for recombinant expression of s cerevisiae ino complex in insect cells and complex purification is described in the accompanying paper krietenstein et al briefly multibac technology trowitzsch et al  was applied to generate two baculoviruses carrying coding sequences for s cerevisiae ino xflag rvb rvb arp arphis arp actin taf ies ies ies ies ies ies and nhp which were subcloned into pfbdm vectors and sequence verified by sanger sequencing gatc services at eurofins genomics high five hi insect cells btitnb invitrogen were coinfected with two or three baculoviruses  vv each for expression purposes the recombinantly expressed ino complex and respective ino mutant complexes were purified from insect cells according to tosi haas et al   which resulted in a pure and monodisperse sample shortly cells were resuspended in lysis buffer  mm trishcl ph   mm nacl   glycerol  mm dtt sigmafasttm protease inhibitor cocktail sonified branson sonifier x  s with   duty cycle and output control  and cleared by centrifugation sorvall evolution rc ss rotor  g the supernatant was incubated with antiflag m affinity gel sigmaaldrich and centrifuged for  min at  g and  c figure d  one nauthorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  https doiorg doiorg  terminal deletion mutant ino n  deletion of the first  amino acids of the n terminus of ino and two ino xflag nhp module mutants nhp ino complex without ies ies ies and nhp but with ino nterminus and hmgii  figure g  were generated and integrated into baculoviruses using multibac technology expression and purification of mutant ino complexes was carried out as described above the ino core complex from chaetomium thermophilum equivalent to the s cerevisiae nterminal deletion mutant was essentially purified as described in expression and purification of fulllength chd and fact hi cells  ml were grown in esf media expression systems and infected with v virus for fulllength chd tagged with a nterminal his tag followed by a mbp tag and a tobacco etch virus protease cleavage site or fact spt carries an nterminal his tag followed by an mbp tag and a tobacco etch virus protease cleavage site for protein expression cells were grown for  hours at  c and subsequently harvested by centrifugation  g  c  min supernatant was discarded and cell pellets resuspended in lysis buffer  mm nacl  mm nahepes ph    vv glycerol  mm dtt  mm imidazole ph   gml leupeptin  gml pepstatin a  mgml pmsf  mgml benzamidine resuspended cells were snap frozen and stored at  c all protein purifications were performed at  c frozen cell pellets were thawed and lysed by sonication lysates were cleared using centrifugation  g  c  min and  g  c  min the supernatant containing chd was filtered with m syringe filters millipore and applied onto a ge histrap hp  ml ge healthcare the column was washed with  column volumes cv lysis buffer  cv high salt buffer  m nacl  mm nahepes ph    vv glycerol  mm dtt  mm imidazole ph   gml leupeptin  gml pepstatin a  mgml pmsf  mgml benzamidine and  cv lysis buffer chd was eluted using a minutes gradient of   elution buffer  mm nacl  mm nahepes ph    vv glycerol  mm dtt  mm imidazole ph   gml leupeptin  gml pepstatin a  mgml pmsf  mgml benzamidine fractions containing chd were pooled and subjected to dialysistev protease digestion for  hours  mm nacl  mm nahepes ph    vv glycerol  mm dtt  mm imidazole with  mg histev protease the dialyzed sample was again applied to a ge histrap hp  ml the flowthrough which contained cleaved tagless chd was concentrated using an amicon millipore  ml  mwco centrifugal concentrator the concentrate was applied to a ge s  pg size exclusion column in  mm nacl  mm nahepes ph    vv glycerol  mm dtt fractions containing chd were concentrated to  m the sample was aliquoted snap frozen and stored at  c fact was purified as above with minor modifications after dialysis the sample was subjected to a tandem ge histrap hp  ml and ge hitrap q  ml columns combination after sample application the columns were washed with lysis buffer and the histrap removed fact was eluted by applying a high salt buffer gradient from   high salt buffer  m nacl  mm nahepes ph    vv glycerol  mm dtt  mm imidazole ph  fractions with fact were applied to a ge s  pg size exclusion column peak fractions with fact were concentrated to a concentration of  m aliquoted snap frozen and stored at  c expression and purifications of iswa isw and fun tandem affinity purification of iswa tapioc and fun tapfun was performed as follows cultures were grown in ypd media harvested cells were washed once with water the cells were lysed in buffer e  mm nahepes ph   mm nacl   glycerol   tween and  mm dtt and protease inhibitors by grinding in the presence of liquid nitrogen lysates were clarified at  g at  c for  h cleared lysates were incubated with iggsepharose ge healthcare at  c for  h and eluted by tev protease invitrogen cleavage at  c overnight the elutions were incubated with calmodulin affinity resin agilent technology in buffer e plus  mm cacl at  c for  h and eluted in buffer e plus  mm egta isw flagisw was purified as follows cleared lysate was incubated with antiflag m affinity gel sigmaaldrich at  c for  h and eluted with  mgml x flag peptide sigmaaldrich ebuffer  mm nahepes ph   mm nacl   glycerol   tween and  mm dtt was used during the entire purification purified proteins were concentrated with vivaspin concentrators sartorius and dialyzed against ebuffer with  mm dtt subunit compositions were confirmed by sdspage  figure sa  and mass spectrometry expression and purification of s cerevisiae reb purification of s cerevisiae reb was essentially carried out as described in  briefly using by genomic s cerevisiae dna the coding sequence for reb was amplified by pcr and cloned into petb novagen via infusion cloning clontech with a streptavidin tag at the c terminus correct sequences were verified via sanger sequencing gatc services at eurofins genomics expression plasmids were transformed into bl de cd  cells three liters of lb medium supplemented with  mgl ampicillin were inoculated with  ml preculture cells were grown at  c to an od of  wpa co cell density meter induction was carried out by addition of iptg to a final concentration of  mm cells were grown overnight at  c harvested by centrifugation  rpm sorvall evolution rc and stored at  c cells were resuspended in lysis buffer  mm trishcl ph   mm nacl   glycerol  mm dtt   sucrose and protease inhibitor sigmafast tm protease inhibitor cocktail  sonicated branson sonifier   min at   duty cycle and output control  and cleared by centrifugation sorvall evolution rc ss rotor  g the supernatant was dialyzed over night against  l low salt buffer  mm khepes ph   mm kcl   glycerol  mm mgcl  mm dtt cation ion exchange chromatography hitrap sp hp  ml elution buffer  mm khepes ph   m kcl   glycerol  mm mgcl  mm dtt followed by size exclusion chromatography superdex   buffer  mm khepes ph   mm kcl   glycerol  mm mgcl  mm dtt were used for purification peak fractions were analyzed by coomassie sdspage fractions containing reb were pooled concentrated and stored at  cpreparation of mononucleosomes with recombinant human octamers canonical human histones were provided by the histone source protein expression and purification pep facility at colorado state university lyophilized individual human histones were resuspended in  m guanidinium chloride mixed at a fold molar excess of authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint figure s associated with figures  and  a sdspage analyses of purified remodeler complexes b composite plots as in figure d for individual replicates and the indicated combinations of remodeler reb and nucleosome density no remodeler denotes absence of remodeler c composite plots aligned at in vivo  nucleosome positions left reb middle or bamhi right sites for mnaseseq analysis of sgd chromatin assembled at high nucleosome density and incubated with the indicated remodelers as in figure a authorfunder all rights reserved no reuse allowed without permission supplementary figures the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint figure a but for indicated combinations of remodeler and nucleosome density no remodeler denotes absence of remodeler no remodeler denotes absence of reb tracks show merged data from replicates shown in figures sb sa reb sites were sorted into groups  to  number of sites per group indicated according to orientation of reb sites relative to one or two genes b as figure c  but averages and standard deviation of values for upand downstream lengths are shown for same groups as in panel a and for all antireb slimchip sites that contain a reb pwm light blue c composite plots left and heat maps right of mnaseseq analysis of in vivo chromatin or sgd chromatin reconstituted with the indicated nucleosome density and incubated with recombinant wt ino complex data are aligned at in vivo  nucleosome positions and heat maps are sorted from top to bottom by increasing nfr length traces with indicated replicate number n represent merged data for all replicates positions of     nucleosomes of the in vivo pattern are labeled d composite plots as in panel c right but for individual replicates and the indicated remodelers no remodeler denotes absence of remodeler no remodeler authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint authorfunder all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint \n","name":"stdout"}]},{"metadata":{},"cell_type":"markdown","source":"***Remove Stopwords***"},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.feature_extraction.text import TfidfVectorizer\n\n\nwith open('../input/stopwords/englishStopwords.txt', 'r') as f:\n    myLists = [line.strip() for line in f]\n\n               \nvectorizer = TfidfVectorizer(stop_words=myLists)\nX = vectorizer.fit_transform(docs)","execution_count":18,"outputs":[{"output_type":"stream","text":"/opt/conda/lib/python3.6/site-packages/sklearn/feature_extraction/text.py:385: UserWarning: Your stop_words may be inconsistent with your preprocessing. Tokenizing the stop words generated tokens ['ain', 'aren', 'couldn', 'didn', 'doesn', 'don', 'hadn', 'hasn', 'haven', 'isn', 'll', 'mon', 'shouldn', 've', 'wasn', 'weren', 'won', 'wouldn'] not in stop_words.\n  'stop_words.' % sorted(inconsistent))\n","name":"stderr"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.model_selection import train_test_split\n\n# test set size of 20% of the data and the random seed 42 <3\nX_train, X_test = train_test_split(X.toarray(), test_size=0.2, random_state=42)\n\nprint(\"X_train size:\", len(X_train))\nprint(\"X_test size:\", len(X_test), \"\\n\")","execution_count":19,"outputs":[{"output_type":"stream","text":"X_train size: 1057\nX_test size: 265 \n\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.cluster import KMeans\n\nk = 10\nkmeans = KMeans(n_clusters=k, n_jobs=4, verbose=10)\ny_pred = kmeans.fit_predict(X_train)","execution_count":20,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_train = y_pred","execution_count":21,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"y_test = kmeans.predict(X_test)","execution_count":22,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"outerlist = []\nwhile len(outerlist) < k:\n    outerlist.append([])","execution_count":23,"outputs":[]},{"metadata":{"trusted":true},"cell_type":"code","source":"for x in docs:\n    Y = vectorizer.transform([x])\n    prediction = kmeans.predict(Y)\n    outerlist[int(prediction)].append(x)\n\norder_centroids = kmeans.cluster_centers_.argsort()[:, ::-1]\nterms = vectorizer.get_feature_names()\n\n\nindex = 0\n\n\nfor iter in outerlist:\n    print(\"DOCUMENTS GROUP %d\" % index)\n    print(iter[:1], sep=', ')\n    print(\" \")\n    \n    print(\"-----------------\")\n\n\n    print(\"GROUP DESCRIPTIVE KEYWORDS\" )\n    for ind in order_centroids[index, :10]:\n        print(' %s' % terms[ind]),\n    index = index + 1\n    print(\"-----------------\")\n\n","execution_count":24,"outputs":[{"output_type":"stream","text":"DOCUMENTS GROUP 0\n['introduction the coronavirus disease  covid caused by severe acute respiratory syndrome coronavirus  sarscov is emerging as a major infectious disease epidemic globally initially detected in a cluster of patients with unexplained pneumonia in wuhan hubei province of china in early december  sarscov rapidly spread not only within china but also around the globe within just three months as of mar    confirmed cases and  deaths have been reported in  countries   although the natural reservoir of sarscov is still unknown early confirmed cases are strongly associated with exposures to wild animals in the huanan seafood wholesale market   and sustained humantohuman transmission is observed particularly among close contacts   due to the surging number and rapid spread of covid world health organization who has increased the risk assessment of covid to very high at the global level on feb  as one of the most developed and commercialized cities in china shenzhen is the largest migrant city where over  of its population  million are migrants the risk of case importation in shenzhen was therefore high especially because the covid epidemic occurred around the chinese lunar new year holiday season during which intracity mobility of the migrant population was extremely high since the first case of covid in shenzhen was confirmed on jan   local authorities activated the highest level of emergency response to the disease although there have been several studies on the transmission epidemiology and clinical symptoms of covid in wuhan     data from other epidemic areas are still lacking   in particular information and knowledge of covid from a migrant city with high population mobility like shenzhen can inform effective prevention and control strategies in other similar settings as such we investigated the epidemiological and clinical characteristics of all  cases that were confirmed in shenzhen  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint as of feb   data collection we conducted a retrospective study of the epidemiological and clinical characteristics of covid cases in shenzhen from jan  to feb   covid case was defined in accordance with the who interim guidelines and the national guidelines in diagnosis and treatment scheme for covid sixth edition    sputum blood bronchoalveolar lavage fluid nasopharyngeal swab or oropharyngeal swab were collected from each patient and tested using realtime polymerase chain reaction rtpcr for sarscov a cycle threshold value less than  was defined as pcr test positive virus detection was first done by hospitals or districted centers for disease control and prevention cdc and positive results were further confirmed by guangdong provincial cdc before jan   or shenzhen cdc after jan   for each confirmed case detailed epidemiological investigations were conducted by local epidemiologists and public health workers all confirmed cases were immediately reported to the national infectious disease information system for covid which was amended as a class b notifiable infectious disease on jan  we compiled the epidemiological and clinical data from both local epidemiological investigation reports and the national infectious disease information system patients information including sociodemographic characteristics exposure history close contacts timelines of illness onset medical visit hospitalization and pcr confirmation symptoms and clinical outcomes was extracted to construct a dataset with no personal identity all confirmed covid cases in shenzhen were included in the current analyses with no prespecified exclusion criteria data collection in the epidemiological investigation was part of the continuing public health investigation of an emerging outbreak and therefore the  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity  study definition patients were defined as having wuhan exposure if they were wuhan residents or visited wuhan within the past  days before symptom onset and were defined as having shenzhen exposure if they had not left shenzhen within the past  days before symptom onset while all others exposed elsewhere were defined as having exposure elsewhere in mainland china other than wuhan or shenzhen based on the national guidelines in diagnosis and treatment scheme for covid  the seriousness of clinical presentations of covid cases was categorized as mild moderate severe and critical mild cases were those with virological confirmation but without an evidence of having pneumonia moderate cases were mild cases at the same time with a diagnosis of pneumonia severe was defined when one of the following criteria was met dyspnea respiratory frequencyminute blood oxygen saturation  paofio ratio  andor lung infiltrates  within  hourscritical cases were defined when patients had respiratory failure andor septic shock andor multiple organ dysfunctionfailure in addition we categorized cases into selfidentified and surveillanceidentified ones the former referred to covid cases who were identified when they sought medical care at the hospitals while the latter referred to cases who were identified through active surveillance efforts including screening of close contacts of the confirmed patients and recent travelers from hubei fever monitoring at airport train station docks and highway checkpoints and registration and report of fever by community workerswe defined a cluster as occurrence of two or more confirmed cases in a sociallyclose setting such as a family a school or a company within the past  days which may be caused by humantohuman transmission through close contacts within such setting or infection via a common external exposure other cases not from a cluster were defined as scattered cases ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint statistical analysis demographic and clinical characteristics as categorical variables were presented as numbers and percentages while continuous variables were presented as means and standard deviations or  confidence intervals ci or medians and interquartile ranges iqr if appropriate intergroup differences in the characteristics were tested by using pearsons   test or fishers exact test for categorical variables and by using students ttest or analysis of variance for continuous variables showing a normal distribution and kruskalwallis and wilcoxon tests for continuous variables with nonparametric distribution the incubation period was estimated by using a previously described parametric accelerated failure time model  patients with detailed information on the time of exposure the date of illness onset or the first time of presentation were included for this analysis we fitted lognormal gamma and weilbull distributions using markov chain monto carlo in a bayesian framework  we estimated the serial interval by using the time difference of illness onset between the infector and infectee initial reproductive number was estimated by using the best fit model based on date of illness onset of the early jan  local exposed cases without relation to the imported cases and the estimated serial interval of covid logistic regression models were applied to identify factors associated with the clinical severity of covid all statistical tests and analyses of the incubation period serial interval and initial reproductive number were performed in r software r foundation for statistical computing results a total of  cases had been confirmed as of feb   in shenzhen in terms of potential source of exposure   were exposed in wuhan   were exposed in shenzhen and   were exposed elsewhere table   in addition   cases were identified by surveillance and   cases occurred in  clusters common  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity which was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint exposure or secondary transmission none of the patients had been to the huanan seafood wholesale market within the past  days before illness onset which was initially thought to be the index location of zoonotic infections of sarscov that started the covid epidemic  days assuming the incubation time followed a lognormal distribution estimates from models with other distributions gamma and weilbull were  to  days see supplemental materials using information on the date of illness onset from  pairs of infector and infectee we estimated the mean serial interval to be   ci  days we estimated the initial reproductive number to be   ci  using information from the cases that exposed in shenzhendemographic characteristics clinical symptoms and outcomes were presented in table  the mean age was  years old and   cases were female patients with wuhan exposure were older than those with exposure in shenzhen or elsewhere  vs    ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint  httpsdoiorg doi medrxiv preprint p patients identified by surveillance were younger than those selfidentified cases  vs  p in total there were   pediatric patients aged below  years of whom  were identified by surveillance and  were cases from clustersthe most common symptoms of covid were fever  dry cough  and myalgia  patients who were exposed in wuhan  identified by surveillance  or scattered  were more likely to have fever p patients with shenzhen exposure were less likely to have dry cough  than patients with wuhan exposure  and elsewhere exposure  they were also less likely to have myalgia  than those with wuhan exposure  patients who were identified by surveillance were less likely to have symptoms than the selfidentified such as fever  vs  dry cough  vs  myalgia  vs  sore throat  vs  and headache  vs  figure a  generally surveillanceidentified patients had shorter time intervals between illness onset the first medical visit hospital admission and pcr confirmation than those selfidentified patients with the estimates to be  vs  days  vs  day  vs  day respectively p for all table   figure b  the scattered cases had a longer time interval from the first medical visit to hospital admission than the clustered cases  vs  day p table  figure c most patients had mild   or moderate   conditions on admission and only   were severe or critical cases  severe pneumonia and  critical  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint shenzhen the proportion of severe or critical cases in shenzhen  was significantly lower than that reported by guan et al in patients across china   in our study older age breathing problems and diarrhea were correlated to the clinical severity on admission other studies also suggested that these factors were associated with poorer outcomes in patients in wuhan and elsewhere    furthermore the longer time from the first medical visit to hospital admission was associated with higher clinical severity on admission while surveillance identification was associated with lower clinical severity thus the low proportion of severe or critical cases in our study might be explained by the large number of cases identified by surveillance and the short time interval from the first medical visit to hospital admission of note   of the severe or critical cases were selfidentified which might be due to delayed hospital admission through active surveillance efforts such as screening of highrisk populations a large proportion of covid patients were identified at the early stage of their illness thus decreasing the possibility of progression to a severe illness the hospital fatality rate of covid in shenzhen  was much lower than  reported from wuhan  and also lower than  reported in a countywide study  since data on clinical outcomes were censored in our study the hospital fatality rate may be underestimatedalthough our study had major strengths such as populationwide case identification in a major migrant city a large sample size and complete profiling of epidemics along the timeline of population interventions there were several limitations that should be acknowledged first since some patients were still under treatment so far we could not yet report the complete data of the disease progression on severity or assess predictors of clinical outcomes second important information such as timelines of possible exposure illness onset and medical visits was selfreported in the epidemiological investigations which might be subject to recall bias third we did not have enough information on comorbid conditions  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint  aor adjusted or the model is constructed using the stepwise logistic regression entry probability  removal probability  with variables including age fever cough with phlegm sore throat headache breath problem diarrhea days from illness onset to the first medical visit days from the first medical visit to admission identification mode and cluster ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint figure  the epidemic curve of covid cases by the date of illness onset and pcr confirmation in shenzhen  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint discussion in this study we reported the epidemiological and clinical characteristics of confirmed covid cases in shenzhen a large migrant city with the highest gdp per capita in mainland china our results can be used to improve the prediction of transmission risk design and implementation of intervention measures and strategies and assessments of intervention effectiveness in similar settingsthe epidemic curve of covid in shenzhen had practical implications overall the rapid increase of cases in shenzhen was interrupted by the massive control measures implemented since jan  and followed by a sustained downward trend the majority of the cases were imported and the local transmission was limited furthermore most cases were selfidentified at the beginning while cases were more likely to be identified by surveillance  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity fever and cough were the most common symptoms of covid on hospital admission in shenzhen which was in accordance with findings from a study that included  patients from  provinces of china   this countywide study also reported that the proportion of patients with fever increased  after hospital admission although the proportion  on admission was lower than that in our study   the overall proportion of patients with cough was lower than that of other study   meanwhile more patients had dry cough  than cough with phlegm  in our study the main symptoms were significantly different regarding different exposure sources patients with shenzhen exposure were less likely to have fever and cough compared to those with wuhan exposure in addition the patients in our study were less likely to have symptoms compared to those patients reported in wuhan  interestingly surveillanceidentified patients showed fewer symptoms at the time of identification than those selfidentified patients in our studythis might be because active surveillance identified cases at an early stage of covidour study reported preliminary findings on the clinical severity of covid in  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity acknowledgments we acknowledge all the staff from the department of communicable diseases control and prevention in shenzhen center for disease control and prevention we also acknowledge joseph wu peng wu and jin zhao for editing and proofreading of this manuscript ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity   elsewhere exposed refers to cases have been exposed in cities in mainland china other than wuhan or shenzhen ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n patients\n medrxiv\n preprint\n covid\n granted\n license\n display\n sarscov\n perpetuitythe\n cases\n-----------------\nDOCUMENTS GROUP 1\n['introduction in december  a novel coronavirus disease covid broke out in wuhan china which caused by severe acute respiratory syndrome coronavirus  sarscov [ ]  the covid had been affirmed to have humantohuman transmissibility [] and has now spread worldwide which raised high attention not only within china but internationally the world health organization who reported that there are   confirmed cases and  deaths in globally until march   []  and evaluated as global pandemic on the same day [] in retrospect studies the outbreak of severe acute respiratory syndrome sars in guangdong in  gradually faded with the warming weather coming and was basically ended until july []  it has been documented that the temperature and its variations might have affected the sars outbreak []  a study in korea found that the risk of influenza incidence was significantly increased with low daily temperature and lowhigh relative humidity a positive significant association was observed for diurnal temperature range dtr []  moreover temperature [] and dtr [] have been linked to the death from respiratory diseases a study demonstrated that absolute humidity had significant correlations with influenza viral survival and transmission rates []  couple of studies reported that the covid was related to the meteorological factors which decreased with the temperature increasing [ ]  but their effects on the mortality had not been reported therefore we assume that the weather conditions might also contributed to the mortality of covidas the capital of hubei province and one of the largest city in central china wuhan is located in the middle of the yangtze river delta which has a typical subtropical humid monsoon climate with cold winters and warm summers []  the average annual temperature and rainfall are  c c and  mm mm respectively []  besides wuhan owns an area of  km  and a population over  million as of  []  so far this covid has caused  deaths globally  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint and caused  deaths in wuhan although the deaths may be affected by many factors the present is to explore the effect from meteorological angel with generalized additive model gam data collection data from  january to   calculation of absolute humidity absolute humidity was calculated according to the previous study and was measured by vapor pressure vp []  the density of water vapor or absolute humidity [ v gm  ] is the mass of moisture per total volume of air it is associated to vp via the ideal gas law for the moist portion of the airwhere e is vapor pressure vp r v is the gas constant for water vapor [ jkg k] and t is the daily ambient temperature k vp is a commonly used indicator of absolute humidity and is calculated from ambient temperature and relative humidity using the clausiusclapeyron relation []  briefly we first calculated the saturation vapor pressure [e s t mb] from daily ambient temperature using the following equationwhere e s t   denotes saturation vapor pressure at a reference temperature t   k which equals to  mb l denotes the latent heat of evaporation for water  kjkg r v is the gas constant for water vapor [ jkg k] t denotes daily ambient temperature k then vp pa was calculated by combining the e s t calculated using eq  with relative humidity rh statistical methods a descriptive analysis was performed we used a generalized additive model gam to analyze the associations between meteorological factors temperature dtr relative humidity and absolute humidity and the daily mortality of covid the core analysis was a gam with a quasipoisson link function based on the previous studies [ ]  we first built the basic models for mortality  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint outcomes without including air pollution or weather variables we incorporated smoothed spline functions of time which accommodate nonlinear and nonmonotonic patterns between mortality and time thus offering a flexible modeling tool then we introduced the weather variables and analyzed their effects on mortality akaikes information criterion was used as a measure of how well the model fitted the data consistent with several recent timeseries studies [ ]  the penalized smoothing spline function was applied to control the effects of confounding factors such as time trends dayofweek and air pollution the core gam equation iswhere t is the day of the observation ey t  is the expected number of daily mortality for covid on day t  is the intercept is the regression coefficient x t is the daily level of weather variables on day t s denotes the smoother based on the penalized smoothing spline based on quasi akaike information criterion qaic the  degrees of freedom df was used for time trends and  df for air pollutants temperature and relative humidity dow is a categorical variable indicating the date of the weekafter establishing the core model we also considered the lag effects of weather conditions on death of covid and examined the potentially lagged effects ie single day lag from lag  to lag  and multipleday average lag from lag  to lag  []  the exposure and response correlation curves between weather variables and covid mortality were plotted using a spline function in the gam we also performed a sensitivity analysis by changing the df of the penalized smoothing spline function from  to  for calendar time and from  to  for temperature and humidityall of the statistical analyses were twosided and at a  level of significance all analyses were conducted using r software version  with the gam fitted by the mgcv package version  the effect estimates were expressed as the percentage changes and their  confidence intervals cis in daily mortality of covid associated with per  unit increase in weather variables  correlation between covid mortality and meteorological factors and air pollutants the correlation results between mortality counts of covid meteorological measures and air pollutant concentrations were presented in table   the mortality counts of covid was negatively associated with relative humidity r   pm  r   and pm  r   a positive  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint association with covid mortality was observed for dtr r   and so  r   and covid mortality at current day generally the curves tended to be not associated with covid mortality for dtr but were strongly positive for temperature in addition the curves associated with relative humidity and absolute humidity presented similar linear trends which indicated that the higher level of humidity might cause decrease in the covid mortality  discussion the study was conducted in wuhan during  january  to  february  in order to investigate the association between weather variables and daily death number of covidsignificant positive effect of dtr on the daily mortality counts of covid was first identified through our study it is also found that a significant negative association with covid mortality is observed for temperature and absolute humidity the results indicate that the effects of dtr should also be paid attention on the mortality of covidour study demonstrated a negative association between covid case fatality and temperature while a positive association for dtr the results are consistent with the following investigationsrespiratory diseases mortality increased with decreasing temperature [] and was strongly associated with low temperature [] while another study found that both cold and heat effects may have adverse impact on respiratory mortality []  otherwise the study conducted in  east asian cities showed that increased dtr was associated with increased risk of mortality for respiratory and cardiovascular diseases []  in the cold season the cumulative relative risk of nonaccidental respiratory and cardiovascular death increased at high dtr values in tabriz []  a timeseries study done in shanghai on the effect of dtr and daily chronic obstructive pulmonary disease copd mortality showing that each c elevation in the day moving average for dtr accounted for  of increased risk of copd mortality []  a review for cold exposure and immune function reported that lower temperature may repress the immune function []  particularly our previous finding suggested that the phagocytic function of pulmonary alveolar macrophages declined under cold stress in vitro experiment []  breathing cold air can lead to bronchial constriction which may promote susceptibility to pulmonary infection []  additionally since sarscov is sensitive to heat and high temperature makes it difficult to survive not to mention the beneficial factors for virus transmission like indoor crowding and poor ventilation in cold days []  also cold temperature has been discovered to be associated with the reduction of lung function and increases in exacerbations for people with copd []  in addition dtr represents a stable measure of temperature which is an indicator of temperature variability to evaluate effects on human health including mortality and  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint morbidity []  besides abrupt temperature changes may add the burden of cardiac and respiratory system causing cardiopulmonary events and high dtr levels may be a source of environmental stress []  our findings highlight a need to strengthen the awareness of the effects of temperature and dtr on covid mortalityresearchers confirmed that respiratory infection was enhanced during unusually cold and low humidity conditions []  indicating low humidity may be also an important risk factor for respiratory diseases worse a study with data of more than  years found that humidity was an important determinant of mortality and lowhumidity levels may cause a large increase in mortality rates potentially by influenzarelated mechanisms []  similar to a study carried out in the united states []  consistent to these finding our results also indicated that the risk of dying from covid decreased only with absolute humidity increasing breathing dry air could cause epithelial damage andor reduction of mucociliary clearance and then lead to render the host more susceptible to respiratory virus infection the formation of droplet nuclei is essential to transmission but exhaled respiratory droplets settle very rapidly at high humidity so that it is hard to contribute to influenza virus spread []  moreover the transmission of pandemic influenza virus is the most efficient under cold dry conditions []  and influenza virus survival rate increased markedly in accordance with decreasing of absolute humidity []  which may be similar to coronavirus therefore the increase of covid mortality may also be related to the lower humidity in winterhowever many limitations should not be ignored firstly there are some other factors may affect the covid mortality such as population migration government intervention and so on secondly ecologic timeseries study design was adopted in the study which might exist ecologic fallacyfurthermore it is difficult to obtain the meteorological and air pollution data at the individual levelnevertheless this study showed that dtr might be associated with the death risk of covid in wuhan which deserves further investigation conclusion this is the first study to investigate the effects of temperature dtr and humidity on daily mortality of covid in chinese population by utilizing publicly accessible data on covid statistics and weather variables our finding shows that daily mortality of covid is positively association with dtr but negatively with relative humidity in summary this study suggests the temperature variation and humidity may be important factors affecting the covid mortality gam generalized additive model ci confidence interval rr relative risk ethical approval and consent to participate not applicable consent for publication not applicable ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint availability of supporting data the datasets used andor analyzed during the current study are available from the websites competing interests the authors declare that they have no conflict of interest funding this work was supported by the novel coronavirus disease science and technology major project in gansu province ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuity author contributions is the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint fig  the exposureresponse curves of meteorological factors and covid daily mortality counts in wuhan china from  january to  february  the xaxis is the concurrent day meteorological data yaxis is the predicted log relative risk rr is shown by the solid line and the dotted lines represent the  confidence interval ci covid coronavirus disease dtr diurnal temperature range ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint  ccbyncnd  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg  doi medrxiv preprint  ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n covid\n preprint\n medrxiv\n patients\n license\n granted\n display\n data\n cases\n perpetuitythe\n-----------------\nDOCUMENTS GROUP 2\n['introduction sarscov previously named ncov is a new coronavirus causing coronavirus disease  covid  which first emerged in december   as of  march  based on the data provided by the world health organization    people all over the world have been infectionconfirmed and  people have died human immunodeficiency virus hiv is one of another most dangerous virus and causes acquired immunodeficiency syndrome aids according to the world health organization who there were  million people living with hiv  early detection of such pathogen infections during seroconversion window will facilitate early intervention test and treat which in turn may reduce disease transmission riskpolymerase chain reaction pcr method is the most commonly used technology for pathogen nucleic acid detection and has been considered as a gold standard for disease diagnostics due to high sensitivity and specificity    however it typically relies on expensive equipment and welltrained personnel all of which is not suitable for point of care diagnostic application in recent decades several isothermal amplification methods such as recombinase polymerase amplification rpa   loopmediated isothermal amplification lamp   have been developed as attractive alternatives to conventional pcr method because of their simplicity rapidity and low cost however there is still a challenge to apply it to develop a reliable poc diagnostics for clinical applications due to nonspecific signals eg falsepositive   recently rnaguided crisprcas nucleasebased nucleic acid detection has shown great promise for the development of nextgeneration molecular diagnostics technology due to its high sensitivity specificity and reliability   for example some cas nucleases eg casa casb and casa perform strong collateral cleavage activities in which a crrnatargetbinding activated cas can indiscriminately cleave surrounding nontarget singlestranded nucleic acids      by combining with rpa preamplification cas and casa have respectively been used to develop sherlock specific highsensitivity enzymatic reporter unlocking system  and detectr dna endonucleasetargeted crispr trans reporter system  for highly sensitive nucleic acid detection apart from rpa method some crisprcasbased nucleic acid sensors utilized the lamp and pcr approaches for instance the crisprcasbassisted holmesv platform  however these crisprcasbased detection methods typically require separate nucleic acid preamplification and multiple manual operations which undoubtedly complicates the procedures and brings about contaminationsin this study we reported an allinone dual crisprcasa termed aiodcrispr assay for rapid ultrasensitive specific and visual detection of nucleic acid dual crrnas are introduced to initiate highly efficient crisprbased nucleic acid detection in our aiodcrispr assay all components for nucleic acid amplification and crispr detection are thoroughly mixed in a single onepot reaction system and incubated at a single temperature eg  c eliminating the need for separate preamplification and amplified product transferring as application examples the aiodcrispr assay was engineered to detect severe acute respiratory syndrome coronavirus  sarscov  and human immunodeficiency virus type  hiv  since sarscov and hiv are retrovirus we evaluated the performance of our aiodcrispr assay by detecting both of their dna and rna especially the test results of our aiodcrispr assay can be directly visualized by naked eye therefore we anticipate that the aiodcrispr assay will facilitate crisprbased nextgeneration molecular diagnostics towards pointofcare applications figure a  the aiodcrispr assay system used a pair of casacrrna complexes generated by two individual crrnas to bind two corresponding sites which are close to the recognition sites of primers in the target sequence the casacrrna complexes were separately prepared prior to being loaded into the solution containing two rpa primers ssdnafq reporters recombinase singlestranded dna binding protein ssb stranddisplacement dna polymerase and target sequences when incubating the aiodcrispr system in one pot at c rpa amplification is first initiated and exposes the binding sites of the casacrrna complexes due to the strand displacement once the casacrrna complexes bind the target sites casa endonuclease is activated and cleaves the nearby ssdnafq reporters to produce fluorescence similarly rpa amplified products are also the substrates to undertake the same process and continuously trigger crisprcasabased collateral cleavage activity previous study   has demonstrated that this collateral cleavage activity is irrelevant to target strand cleavage in crisprcasa system therefore target sequences for the aiodcrispr assay are not limited by the casas protospacer adjacent motif pam   aiodcrispr assay system as shown in to evaluate and develop our aiodcrispr assay we prepared and tested eight reaction systems reactions   with various components figure b i  the ssdnafq reporter was a  nt oligonucleotide ttatt labelled by  fam fluorescein fluorophore and  iowa black  fq quencher after incubated at c for  min only reaction   containing the target sequence dual crrnas casa and the rpa mix produced superbright fluorescence signal which could be directly visualized in a led blue or uv light illuminator interestingly even under ambient light environment without any excitation a color change from orangeyellow to green was observed in the reaction tube   by naked eyes furthermore the assay products were verified by denaturing polyacrylamide gel electrophoresis page as shown in figure b ii a strong band with shorter dna size was observed only in the lane of the reaction   which resulted from the cleaved ssdnafq reporter with bright fluorescence on the contrary for other reaction systems only weak bands with slightly longer dna sizes were observed in their corresponding lanes due to fluorescence quench of the intact uncut ssdnafq reporters in addition in realtime fluorescence detection only the reaction   showed an elevated fluorescence signals that saturated within  min  figure biii  thus these results show that our aiodcrispr assay provides a simple rapid onepot approach for targetspecific nucleic acid detectionsince rpa amplification reaction is initiated after adding mgoac  we are interested in knowing if the nucleic acid amplification was initiated at room temperature during sample preparation in our aiodcrispr assay system we prepared two aiodcrispr solutions one positive and one negative and allowed them to remain at room temperature for  min as shown in figure s  no significant fluorescence change between positive and negative samples was observed in the aiodcrispr solutions at room temperature on the contrary there was an obvious fluorescence increasing after just onemin incubation at c  figure  s  the fluorescence signal was eventually saturated at  min after incubated at c therefore our aiodcrispr assay system is mainly triggered only after reaction temperature arrives at c optimization of aiodcrispr assay collateral cleavage of ssdnafq reporter by the casa is triggered by the binding of crrna to target sites   thus we hypothesized that higher binding opportunities could increase the collateral cleavage activity and eventually improve the detection sensitivity for aiodcrispr to prove this we introduced a pair of crrnas to recognize the corresponding target sites a pucidtamp plasmid containing  bp hiv p gene cdna p plasmid was used as the target and three different designs for primers and crrnas were investigated figure a  as shown in figure b  aiodcrispr with dual crrnas crrnacrrna shows similar fluorescence curve as that with crrna but was much better than using crrna in realtime detection doubling the amount of crrna or crrna actually didnt benefit the detection efficiency on detection sensitivity figure c shows that aiodcrispr with dual crrnas was able to detect as low as  copies of the p plasmid templates within  min while aiodcrispr with single crrna doesnt detect it thus introducing dual crrnas is beneficial to improve the sensitivity of aiodcrispr assaywe further optimized ssdnafq reporters in our the aiodcrispr assay because the reporter concentration is crucial in fluorescence readout as shown in figure sa  higher  ccbyncnd  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint concentration of ssdnafq reporters was used shorter time was taken to reach saturated fluorescence intensity as to threshold time and visual detection the minimal concentration for saturated values was  m figure sbsd  collateral cleavage by activated casa is an activity of cutting nearby ssdnafq reporters   thus increasing ssdnafq reporter concentration can strengthen the fluorescence signals in addition primer concentration also affected the aiodcrisprs efficiency as shown in figure s  the optimal concentration should be  m together introducing dual crrnas and using high concentration of ssdnafq reporters enable efficient aiodcrispr with  m of each primer the pucidtamp plasmid containing  bp hiv p gene cdna p plasmid and the sequences of its primers and crrnas b realtime fluorescence detection of the allinone crisprcasa assay using dual crrnas crrnacrrna or single crrna crrnacrrna crrnacrrna means doubling its amount p the positive reaction with    copies of hiv p plasmids c the crisprcasa assays with dual crrnas crrnacrrna or crrna for the detection of  copies of hiv p plasmids p three replicates ran for each reaction with the plasmid ntc nontemplate control reaction each reaction contained  m ssdnafq fluorescence images were taken after  min incubation hiv detection by aiodcrispr assay to investigate its sensitivity for hiv dna detection we first applied the optimized aiodcrispr assay to detect various copies of hiv p plasmid templates from    to    copies as shown in figure a  the aiodcrispr could detect as low as  copies hiv p plasmid dna in both realtime and endpoint visual detection in  min which was further verified by the denaturing page  figure  b  although incubated for  min the aiodcrispr actually could identify  copies of hiv dna in just  min incubation based on endpoint fluorescence as shown in figure c  which showed that our aiodcrispr assay provides a superfast and ultrasensitive approach for nucleic acid detection c endpoint fluorescence comparison after  min incubation at  c ntc nontemplate control reaction three replicates ran for each reaction or test error bars represent the standard deviations at three replicates n   unpaired twotailed ttest was used to analyse the difference from ntc p   p   p   p   ns not significantnext we applied the aiodcrispr assay to detect hiv rna sequence by adding avian myeloblastosis virus amv reverse transcriptase namely reverse transcription aiodcrispr rtaiodcrispr the hiv rna target is a nt fragment of gag p included gene prepared using t promotortagged rtpcr and t rna polymerasebased transcription  figure sa  the detection region of aiodcrispr was further verified by sanger sequencing figure sb  to achieve highly sensitive hiv rna detection we optimized the amv concentration as shown in figure sc  the rtaiodcrispr performed the highest efficiency with  ul amv when incubated at c furthermore we investigated the sensitivity of our rtaiodcrispr assay using            and    copies of hiv gag rna templates figure a shows that rtaiodcrispr incubated for  min at c was able to consistently detect  copies rna targets in realtime endpoint fluorescence and visual detections under led blue and uv light however visual detection without excitation light only identified    copies rna targets due to unsaturated fluorescence for other reactions additionally the rtaiodcrisprs sensitivity was further confirmed by denaturing page analysis  figure s  in addition to assaying artificial hiv gag rna we also analysed rtaiodcrisprs performance using hiv rna extracted from plasma samples as shown in figure b  realtime rtaiocasarbased detection was able to detect the viral rna extracted from  copiesl hiv plasma sample though displaying low fluorescence change in less than  min however visual rtaiodcrispr detection took relatively long incubation time up to  min to achieve the similar sensitivity the reduced efficiency of rtaiodcrispr likely results from some potential inhibitors or disruptors in the extracts despite this realtime aiodcrispr assay was still comparable to realtime rtpcr assay regarding the same detection sensitivity figure c and s  figure a  a pucidtamp plasmid containing  nt sarscov n gene cdna n plasmid was first prepared as the target to develop our aiodcrispr assay figure b shows that our aiodcrispr assay could detect  copies of sarscov n plasmids in both realtime and visual detections within  min offering a rapid and nearly singlemolecule level sensitive detection to evaluate the detection specificity we tested our aiodcrispr assay using commercially available control plasmids containing the complete n gene from sarscov sarscov _pc sars sarscov_pc and mers merscov_pc as well as the hs_rpp control hs_rpp_pc with a portion of human rpp gene figure c shows that only the reaction with sarscov_pc shows the positive signal in both realtime and visual  ccbyncnd  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint detections demonstrating that our developed aiodcrispr assay possesses high specificity without cross reactions for nonsarscov targets sarscov detection by aiodcrispr assay as shown in next we used t promotortagged pcr and t rna polymerase to prepare sarscov n gene rna sequences to develop the rtaiodcrispr assay figure a  the detection region of rtaiodcrispr was verified by sanger sequencing figure b  as shown in figure c  both realtime and visual aiodcrisprbased detections could detect down to  copies of sarscov n rna targets in  min therefore our aiodcrispr assay is demonstrated to be a rapid highly sensitive and specific sarscov detection method unpaired twotailed ttest was used to analyse the difference from ntc p   p   p   p   ns not significant discussion in this study we describe a new crisprcasa reaction aiodcrispr in which all components are incubated in a single reaction system to rapidly highly sensitively and specifically detect nucleic acids without separate preamplification steps upon the design of aiodcrispr dual crrnas are introduced to improve the sensitivity and highconcentration  ccbyncnd  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint ssdnafq reporters are added to strengthen the detection signals in addition to realtime detection aiodcrispr assay is also beneficial for pointofcare detection because its detection results can be visually judged based on the fluorescence and color change of reaction solutions as proofofconcept assays aiodcrispr has been successfully developed for sarscov and hiv detectioncompared to previously reported crisprbased nucleic acid detection       our versatile and robust aiodcrispr has some distinctive advantages first aiodcrispr system is a true single reaction system all the components are prepared prior to incubation thoroughly circumventing the preamplification of target nucleic acids  physical separation of cas enzyme  or the requirement of specially designed tubes or devices   second aiodcrispr is a true isothermal nucleic acid detection method aiodcrispr is conducted in onestep and onepot format at one temperature eliminating the need of expensive thermocycler in pcr methods and the initial denaturing of dsdna targets in isothermal nucleic acid amplification techniques such as lamp  and sda   third aiodcrisprbased detection is very fast robust highly specific and nearly singlemolecule sensitive in the application examples of hiv and sarscov detections our engineered aiodcrispr without preamplification is able to detect as low as  copies dna targets and  copies rna targets in  min incubation especially in detecting hiv p plasmids aiodcrispr can stably identify  copies in just  min incubation for emerging sarscov in addition to high sensitivity the aiodcrispr also performs high detection specificity compared to the reported realtime rpa   aiodcrispr shows a robust hiv detection with very low background fourth the aiodcrispr enables onestep crisprcasabased rna detection by supplementing amv reverse transcriptase aiodcrispr can be easily developed as onestep rtaiodcrispr to detect rna targets such as hiv and sarscov rnas which facilitates the crisprcasabased rna detection without the preparation of cdna on detecting hiv plasma samples aiodcrispr is comparable to realtime rtpcr assayin summary aiodcrispr using a true single reaction system is a rapid robust highly specific and nearly singlemolecule sensitive isothermal nucleic acid detection method this straightforward crisprcasa nucleic acid sensing system has great potential in enabling crisprbased nextgeneration molecular diagnostics towards pointofcare quantitation and digital analysis methods materials oligonucleotides primers ssdnafq reporters pucidt amp plasmids with customized gene sequences and the control plasmids containing the complete n gene from sarscov sars and mers as well as the hs_rpp control were synthesized by or purchased from integrated dna technologies coralville ia the thermal cycling protocol included  min at c for reverse transcription  min at c for initial pcr activation step  cycles of the step cycling  s at c for denaturation  s at c for annealing and  min at  c for extension and  min at c for final extension followed by the meltcurve analysis from  c to  c with  c increment realtime onestep rtpcr assay was conducted in the cfx touch realtime pcr detection system and the plate read was set at the annealing in the step cycling figure s  the aiodcrispr assay at room temperature for  min before incubation and incubated at c for   and  min positive p the reaction with    copies of hiv p plasmids ntc nontarget control reaction the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n reads\n samples\n rna\n sequencing\n preprint\n sarscov\n viral\n detection\n genome\n dna\n-----------------\nDOCUMENTS GROUP 3\n['background similar to outbreaks of other respiratory pathogens      syndromic airport screening at arrival of travellers from regions with a high risk of humantohuman transmission of sarscov is unlikely to prevent a sufficient proportion of infected travellers to prevent global spread    sensitising arriving travellers to the symptoms and risk of sarscov and to encourage selfisolation as well as rapidly seeking for medical assistance via telephone which in turn will trigger tracing and quarantine of contacts may have a more pronounced effect and is currently implemented in many transport hubs however with increasing numbers of infected travellers contact tracing is unlikely to be sustainable for long because of the immensely resourceintensive nature of contact tracing and hence is similarly unlikely to prevent local transmission in the long term  if containment is impossible another key target of the pandemic response is to delay local spread   this will allow additional time for preparation of the health system and mobilisation of additional public health resources delaying local spread will also allow for crucial time to better understand the pathogen and to evaluate effective treatment and prevention measures we aim to estimate the effectiveness of syndromic screening and traveller sensitisation for delaying the onset of sustained sarscov spread in previously unaffected regions outbreak probability model we represent the potential importation of infections by an average number of infected travellers per week   attempting to travel to a specific country or region currently not experiencing an outbreak we assume that individuals with severe symptoms do not attempt to travel though travellers may develop severe symptoms en route upon arrival we assume all infected travellers have the same distribution of potential secondary infections determined by the average number of those infections   and the dispersion of that number   following hartfield and alizon   model of symptom screening and sensitisation we represent syndromic screening effects by the probability of missing an infected traveller   when screening is implemented the average number of infected travellers arriving in a week reduces from  to   and thereby increases   for the scenarios we consider we assume the same baseline assumptions as in quilty et al   ie a syndromic screening sensitivity of  travel duration of  hours and average times from infection to onset of symptoms and from onset to severe symptomshospitalisation as  and  days respectively for those assumptions quilty et al estimate the probability of sarscov infected travellers not being detected at either exit or entry screening as  and as  for exitonly screening  we represent traveller sensitisation as reducing  sensitisation occurs via eg posters and handouts to travellers arriving from high risk regions which increases the likelihood that those travellers if they experience sarscov symptoms will selfisolate on the occurrence of mild symptoms and rapidly report to health care providers who in turn trigger contact tracing as a base case in line with hellewell et al  we assume that these measures to accelerate selfisolation and reporting in the early stages of the sarscov pandemic can reduce the average number of onward transmitting secondary infections by about  calculation of delays to reach outbreak threshold to determine the impact of interventions we calculate the difference in time to outbreak occurrence with and without interventions  scenarios considered we considered three syndromic screening intervention scenarios no screening exitonly and exitandentry screening we further considered two scenarios of traveller sensitisation  and  effectiveness no screening and  sensitisation effectiveness form the nonintervention reference these are studied in the context of either   or  infected travellers per week we assume that  is gamma distributed with an inner  range from  to   and that its dispersion is k  for sensitivity analyses we also investigate alternative scenarios for the dispersion of  and the effectiveness of traveller sensitisation all scenarios and parameters are summarised in table  all analyses were done with r   and can be found on github at httpsgithubcomcmmidscreening_outbreak_delay  ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint    ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  figure   schematic of the air traveller intervention process a proportion of infected travellers will be detected through syndromic exit or entry screening and will immediately be isolated and not cause secondary cases in the yet unaffected destination sensitised travellers will enter the destination but cause fewer secondary cases because of an increased likelihood of selfisolation and rapid careseeking results in the case of the slowest arrival rate  infected traveller per week the combination of traveller sensitisation and exit and entry screening the expected benefit  of simulations predict delays at least as large as this is a delay of  days   days   days table  figure   if  infected travellers per week are simulated the outbreak delay reduces to  days   days   daythe incremental benefit of syndromic entry screening is highly dependent on the effectiveness of exit screening with one infected traveller per week traveller sensitisation and under baseline assumptions of exit screening effectiveness but no entry screening the outbreak is delayed by  days   days   days indicating that additional entry screening adds little in this case table   summary statistics providing the inner  and  confidence intervals and medians all rounded to the nearest day for the estimated number of days an outbreak is delayed given an intervention consisting of a combination of traveller screening and sensitisation and contact tracing comparisons are made to no contact tracing and no screening there are no no screening results at  sensitisation as this is the baseline against which comparisons are to be made an e g  ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint we estimate that with no traveller sensitisation and under baseline assumptions for the effectiveness of syndromic screening at exit and entry in the early stages of the outbreak with  infected traveller per week an outbreak may be delayed by  days    less than  day again this is largely due to exit screening at departure which on its own is estimated to delay the outbreak by  days   days  less than  day as the number of infected travellers increases during the overseas outbreak the delay in onset of a local outbreak through screening declines rapidly specifically if infected traveller numbers approach  per week syndromic screening alone can only delay the outbreak by  day  less than  day ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint similarly we estimate that in the absence of syndromic air traveller screening traveller sensitisation can delay the outbreak by  days   days   days early in the epidemic with  infected traveller per week but delays reduce to  days   day  less than  day for  infected travellers per weekfor sensitivity analyses we varied the effectiveness of traveller sensitisation and the heterogeneity in the number of secondary infections a  reduction in the effective reproduction number through traveller sensitisation followed by rapid case isolation and contact tracing can potentially prevent a local outbreak independent of the number of infected arrivals if the basic reproduction number is smaller than  ie as traveller sensitisation increases and therefore a greater proportion of simulated   values are less than  the proportion of simulated delays that are infinitely long indicating that that specific simulated potential outbreak has been averted increases table s  if traveller sensitisation is assumed to be only  effective then the associated outbreak delay for the base case with one infected traveller per week and syndromic screening is  days   days   days  table  s  if the number of secondary infections is substantially less disperse eg influenzalike outbreak delays decrease by about  as the outbreak becomes less reliant on occasional superspreading events  figure s  if however the number of secondary infections is slightly more disperse eg sarslike then outbreak delays are approximately threefold longeradditional figures in the appendix show the complementary cumulative density functions focusing on either variation with traveller sensitisation  figure s  or arrival rate  figure s  ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  figure   complementary empirical cumulative density functions for the estimated number of days an outbreak is delayed given an intervention consisting of a combination of traveller screening and sensitisation and contact tracing rows correspond to different arrival rates and columns to traveller sensitisation comparisons are made to no contact tracing and no screening there are no no screening results at  sensitisation as this is the baseline against which comparisons are to be made discussion syndromic screening of air travellers at departure andor arrival is unlikely to prevent a sufficient proportion of sarscov infected travellers from entering a yet unaffected country and thereby prevent a local outbreak similarly sensitisation of travellers from highrisk countries to encourage selfisolation and enable accelerated case detection and contact tracing if indeed infected will likely not be able to halt an outbreak indefinitely particularly when many infected  ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  travellers arrive undetected we investigate here how syndromic screening and traveller sensitisation as well as their combination may delay an outbreak of sarscov we find that while there is only about  infected traveller per week syndromic airport screening can only slightly delay an outbreak  days   days  less than  day while traveller sensitisation can have a more pronounced effect be delaying the outbreak by  days   days   days the combination of syndromic screening and traveller sensitisation may more substantially delay an outbreak while the number of infected travellers is low  days   days   days the incremental effect of syndromic entry screening is only notable if exit screening is poor once the weekly number of infected passengers increases to  and above even with the combination of syndromic screening and traveller sensitisation delays of more than a week are unlikely these results are sensitive to a number of key assumptions with increasing r   less heterogeneous r  or less effect of traveller sensitisation the estimated achievable delay quickly becomes negligible we find a potential role for interventions targeting airpassengers to delay major outbreaks of sarscov in previously unaffected regions as long as implemented when there are only a few infected travellers per week we find that syndromic screening on arrival can add to the effect of traveller sensitisation in these early stages of a pandemic syndromic screening can also aid to reduce the number of passengers that would eventually selfreport and then require resourceintensive follow up including contact tracing therefore syndromic screening may have an additional role in helping to sustain control efforts for longer of note however is that syndromic screening at arrival only substantially adds to control efforts if syndromic screening at departure is absent or largely ineffectivewhile our findings may encourage implementation of both syndromic screening on entry and traveller sensitisation in the early stages of the sarscov pandemic it is important to note that these findings are highly sensitive to the underlying basecase assumptions and do not consider the economic implications of large scale air passenger screening and contact tracing with increasing numbers of infected travellers a higher number of secondary infections or a lower heterogeneity thereof or less effective interventions the achievable delay quickly drops down to a few days of delay while all of our assumptions include the best knowledge on sarscov to date there is considerable uncertainty associated with all of these assumptions for example we have assumed recently reported heterogeneity in the individual r   however the reported range of uncertainty includes sarslike and influenzalike which can drastically alter the results some recent yet not peerreviewed estimates would suggest more sarslike or even more overdispersed k which would imply that longer outbreak delays are possible as shown in our sensitivity analysis   we also dont explicitly account for potential asymptomatic transmission however we implicitly do so as both the syndromic screening as well as the contact tracing work that informed our estimates accounted for a small proportion of asymptomatic transmitters who we assume similarly transmit but will not be affected by syndromic screening or sensitisation notably we assumed a constant rate of infected travellers however this may increase rapidly as the epidemic continues to spread exponentially in china and potentially elsewhere there is currently little evidence for an exponential increase in infected travellers to europe as airports in the highest risk regions in china have shut if  ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint indeed infected traveller numbers were to increase exponentially numbers would increase from  to  and  per week within about  and  days respectively assuming r and serial interval of  and estimated delays would decrease accordingly furthermore the results are predicated on a syndromic screening sensitivity of    when reducing the sensitivity to  as used in other reports   delays reduce by about all major airlines have currently suspended flights from mainland china with travel restrictions from iran italy and korea being recently added over the last few weeks only a few sarscov infected travellers have been reported with symptoms within a few days after their arrival  this may suggest that indeed the uk and some parts of europe are still in a situation where airtraveller targeted interventions may substantially delay major local outbreaks however underreporting of cases is likely and with quickly rising case numbers in hong kong singapore japan korea and now iran and italy as well this situation may change rapidly in the coming days and weeksdespite limited evidence that supports its effectiveness   the us has recently banned entry to the us by most foreigners who have recently visited china   in our work we did not investigate such a drastic interruption of air travel while a travel ban for entry with history of travel to all highrisk regions would indeed likely further limit the number of infected travellers entering to those who enter by different means it would also come with substantial economical implications it does also run the risk that travellers arriving in the us despite their travel history would likely not be reached by targeted sensitisation andor that they would avoid selfreporting if symptomatic with potentially dire implications for local spreadin summary we find that targeting airtravellers with syndromic screening at exit or entry and sensitisation for signs of symptoms following their arrival may delay a major outbreak in the early stages of the sarscov outbreak in most countries airtraveller sensitisation and rapid contact tracing protocols are already in place in response to the sarscov outbreak we find that syndromic screening at arrival may further enhance such control efforts while the number of infected passengers is less than about  per week but only in the absence of syndromic screening at departureneither patients nor the public were involved with the design conduct reporting or dissemination plans of our research as this work is a simulation study there are no participants to which we can disseminate the results of this research ccby  international license it is made available under a authorfunder who has granted medrxiv a license to display the preprint in perpetuityis the which was not peerreviewed the copyright holder for this preprint  ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n screening\n travellers\n pro\n traveller\n ncov\n preprint\n sensitisation\n lombardy\n protease\n monitoring\n-----------------\nDOCUMENTS GROUP 4\n['introduction high levels of protein expression and solubility are two major requirements of successful recombinant protein production esposito and chatterjee   however recombinant protein production is a challenging process almost half of recombinant proteins fail to be expressed and half of the successfully expressed proteins are insoluble  httptargetdbrcsborgmetrics  these failures hamper protein research with particular implications for structural functional and pharmaceutical studies that require soluble and concentrated protein solutions kramer et al  hou et al   therefore solubility prediction and protein engineering for enhanced solubility is an active area of research notable protein engineering approaches include mutagenesis truncation ie expression of partial protein sequences or fusion with a solubilityenhancing tag waldo  esposito and chatterjee  trevino martin scholtz and nick pace  chan et al  kramer et al  costa et al   protein solubility at least in part depends upon extrinsic factors such as ionic strength temperature and ph as well as intrinsic factorsthe physicochemical properties of the protein sequence and structure including molecular weight amino acid composition hydrophobicity aromaticity isoelectric point structural propensities and the polarity of surface residues wilkinson and harrison  chiti et al  tartaglia et al  diaz et al   many solubility prediction tools have been developed around these features using statistical models eg linear and logistic regression or other machine learning models eg support vector machines and neural networks hirose and noguchi  habibi et al  hebditch et al  sormanni et al  heckmann et al  z wu et al  yang wu and arnold  in this study we investigated the experimental outcomes of  recombinant proteins expressed in escherichia coli from the protein structure initiativebiology psibiology chen et al  acton et al   we showed that protein structural flexibility is more accurate than other protein sequence properties in predicting solubility craveur et al  m vihinen torkkila and riikonen   flexibility is a standard feature that appears to have been overlooked in previous solubility prediction attempts on this basis we derived a set of  values for the standard amino acid residues and used them to predict solubility we call this new predictor the solubilityweighted index swi swi is a powerful predictor of solubility and a good proxy for global structural flexibility in addition swi outperforms many existing de novo protein solubility prediction tools global structural flexibility performs well at predicting protein solubility we sought to understand what makes a protein soluble and develop a fast and accurate approach for solubility prediction to determine which protein sequence properties accurately predict protein solubility we analysed  target proteins from over  species that were expressed in e coli the psibiology dataset see supplementary fig s and table sa  chen et al  acton et al   these proteins were expressed either with a cterminal or nterminal polyhistidine fusion tag pet_nesg and pet_nesg expression vectors n and  respectively they were previously curated and labeled as protein_soluble or tested_not_soluble seiler et al   based on the soluble analysis of cell lysate using sdspage r xiao et al   a total of  recombinant proteins were found to be soluble in which  of them belong to the pet_nesg dataset both the expression system and solubility analysis method are commonly used costa et al   therefore this collection of data captures a broad range of protein solubility issues  the solubilityweighted index swi is an improved predictor of solubility protein structural flexibility in particular the flexibility of local regions is often associated with function craveur et al   the calculation of flexibility is usually performed by assigning a set of  normalised bfactorsa measure of vibration of calpha atoms see supplementary notesto a protein sequence and averaging the values by a sliding window approach ragone et al  karplus and schulz  m vihinen torkkila and riikonen  smith et al   we reasoned that such sliding window approach can be approximated by a more straightforward arithmetic mean for calculating global structural flexibility see supplementary notes we determined the correlation between flexibility vihinen et al s sliding window approach as implemented in biopython and solubility scores calculated as followswhere is the normalised bfactor of the amino acid residue at the position  and is the b i i l sequence length we obtained a strong correlation for the psibiology dataset spearmans   ccby  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint rho   pvalue below machines underflow level therefore we reasoned that the sliding window approach is not necessary for our purposewe applied this arithmetic mean approach ie sequence composition scoring to the psibiology dataset and compared four sets of previously published normalised bfactors bhaskaran and ponnuswamy  ragone et al  m vihinen torkkila and riikonen  smith et al   among these sets of bfactors sequence composition scoring using the most recently published set of normalised bfactors produced the highest auc score to improve the prediction accuracy of solubility we iteratively refined the weights of amino acid residues using the neldermead optimisation algorithm nelder and mead   to avoid testing and training on similar sequences we generated  crossvalidation sets with a maximised heterogeneity between these subsets ie no similar sequences between subsets we first clustered all  psibiology protein sequences using a  similarity threshold using usearch to produce  clusters with remote similarity see methods and supplementary fig s  the clusters were grouped into  crossvalidation sets of approximately  sequences each manually we did not select a representative sequence for each cluster as about  of clusters contain a mix of soluble and insoluble proteins supplementary fig sc  more importantly to address the issues of sequence similarity and imbalanced classes we performed  bootstrap resamplings for each crossvalidation step fig a and supplementary fig s  we calculated the solubility scores using the optimised weights as equation  and the auc scores for each crossvalidation step our training and test auc scores were    and    respectively showing an improvement over flexibility in solubility prediction mean  standard deviation fig b and supplementary table s the final weights were derived from the arithmetic means of the weights for individual amino acid residues obtained crossvalidation supplementary table s  we observed over a  change on the weights for cysteine c and histidine h residues fig c and  supplementary table s  these results are in agreement with the contributions of cysteine and histidine residues as shown in supplementary fig sb we call the solubility score of a protein sequence calculated using the final weights the solubilityweighted index swi smith et al   the solubility score of a protein sequence was calculated using a sequence composition scoring approach equation  using optimised weights  w instead of normalised bfactors  these scores were used to compute the auc scores for b training and test datasets b training and test performance of solubility prediction using optimised weights for  amino acid residues in a fold crossvalidation mean auc  standard deviation related data and figures are available as supplementary table s and supplementary fig s and s  c comparison between the  initial and final weights for amino acid residues the final weights are derived from the arithmetic mean of the optimised weights from crossvalidation these weights are used to calculate swi the solubility score of a protein sequence in the subsequent analyses filled circles which represent amino acid residues are colored by hydrophobicity kyte and doolittle   solid black circles denote aromatic amino acid residues phenylalanine f tyrosine y tryptophan w dotted diagonal line represents no change in weight see also supplementary table s and fig s auc area under the roc curve roc receiver operating characteristic  arithmetic w mean of the weights of an amino acid residue optimised from  bootstrap samples in a crossvalidation step   ccby  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint to validate the crossvalidation results we used a dataset independent of the psibiology data known as esol niwa et al    this dataset consists of the solubility percentages of e coli proteins determined using an e coli cellfree system n    our solubility scoring using the final weights showed a significant improved correlation with e coli protein solubility over the initial weights smith et al s normalised bfactors [spearmans rho of  p       versus  p      ] we repeated the correlation analysis by removing extra amino acid residues including histags from the esol sequences mrgshhhhhhtdpalra and glcgr at the nand ctermini respectively this artificial dataset was created based on the assumption that histags have little effect on solubility we observed a slight decrease in correlation for this artificial dataset spearmans rho   p    which may be due to the effects of histag in solubility andor the limitations of our approach that may overfit to histag fusion proteinswe performed spearmans correlation analysis for both the psibiology and esol datasets swi shows the strongest correlation with solubility compared to the standard and  protein sequence properties fig  and supplementary fig s respectively  swi also strongly correlates with flexibility suggesting that swi is also a good proxy for global structural flexibility we asked whether protein solubility can be predicted by surface amino acid residues to address this question we examined a previously published dataset for the protein surface stickiness of  e coli proteins levy de and teichmann   this dataset has the annotation for surface residues based on previously solved protein crystal structures we observed little correlation between the protein surface stickiness and the solubility data from esol spearmans rho   p   n   supplementary fig sa  next we evaluated if amino acid composition scoring using surface residues is sufficient optimising only the weights of surface residues should achieve similar or better results than swi as above we iteratively refined the weights of surface residues using the neldermead optimisation algorithm the method was initialised with smith et al s normalised bfactors and a maximised correlation coefficient was the target however a low correlation was obtained upon convergence spearmans rho   p       supplementary fig  sb  in contrast the swi of the fulllength sequences has a much stronger correlation with solubility spearmans rho   p       supplementary fig sc  these results suggest that the fulllength of sequences contributes to protein solubility not just surface residues in which solubility is modulated by cotranslational folding natan et al   to understand the properties of soluble and insoluble proteins we determined the enrichment of amino acid residues in the psibiology targets relative to the esol sequences see methods we observed that the psibiology targets are enriched in charged residues lysine k glutamate e and aspartate d and depleted in aromatic residues tryptophan w albeit to a lesser extend for insoluble proteins supplementary fig sa  as expected cysteine residues c are enriched in the psibiology insoluble proteins supporting previous findings that cysteine residues contribute to poor solubility in the e coli expression system diaz et al  wilkinson and harrison  in addition we compared the swi of random sequences with the psibiology and esol sequences we included an analysis of random sequences to confirm whether swi can distinguish between biological and random sequences we found that the swi scores of soluble proteins are higher than those of insoluble proteins supplementary fig sb  and that true biological sequences also tend to have higher swi scores than random sequences highlighting a potential evolutionary selection for solubility swi outperforms many protein solubility prediction tools to confirm the usefulness of swi in solubility prediction we compared it with the existing tools proteinsol hebditch et al    camsol v sormanni aprile and vendruscolo  sormanni et al   parsnip   deepsol v khurana et al   the wilkinsonharrison model davis et al  harrison  wilkinson and harrison   and ccsol omics agostini et al    we did not include the specialised tools that model protein structural information such as surface geometry surface charges and solvent accessibility because these tools require prior knowledge of protein tertiary fig a  our swi c program is also the fastest solubility prediction algorithm table  fig b and supplementary table s  the wall time was reported at the level of machine precision mean seconds  standard deviation a total of  sequences were chosen from the psibiology and esol datasets related to fig b and supplementary table s see methods   b for swi mean auc  standard deviation was calculated from a fold crossvalidation see methods for other tools no crossvalidations were done as the auc scores were calculated directly from the individual subsets used for crossvalidation c deepsol reports solubility prediction as probability and binary classes the probability of solubility was used to calculate auc and spearmans correlation due to better results auc area under the roc curve na not applicable pdb protein data bank psibiology protein structure initiativebiology roc receiver operating characteristic r s  spearmans rho swi solubilityweighted index s seconds fig  swi outperforms existing protein solubility prediction tools a prediction accuracy of solubility prediction tools using the above crossvalidation sets fig a  for swi the test auc scores were calculated from a fold crossvalidation ie a boxplot representation of fig b  for other tools the test auc scores were calculated directly from the individual subsets used for crossvalidation no crossvalidations were done camsol and ccsol omics are only available as web servers no fill colors b wall time of protein solubility prediction tools per sequence log scale all command line tools were run three times using  sequences selected from the psibiology and esol datasets related data is available as supplementary to demonstrate a use case for swi we developed the soluble domain for protein expression sodope web server see methods and httpstisignercomsodope  upon sequence submission the sodope web server enables users to navigate the protein sequence and its domains for predicting and maximising protein expression and solubility  discussion protein structural flexibility has been associated with conformal variations functions thermal stability ligand binding and disordered regions mauno vihinen  teague  ma  radivojac  schlessinger and rost  yuan bailey and teasdale  yin li and li   however the use of flexibility in solubility prediction has been overlooked although their relationship has previously been noted tsumoto et al   in this study we have shown that flexibility strongly correlates with solubility fig   based on the normalised bfactors used to compute flexibility we have derived a new position and length independent weights to score the solubility of a given protein sequence ie sequence composition based score we call this protein solubility score as swiupon further inspection we observe some interesting properties in swi swi anticorrelates with helix propensity gravy aromaticity and isoelectric point fig c and   suggesting that swi incorporates the key propensities affecting solubility amino acid residues with a lower aromaticity or hydrophilic are known to improve protein solubility trevino martin scholtz and nick pace  niwa et al  kramer et al  warwicker charonis and curtis  han et al  wilkinson and harrison   consistent with previous studies the charged residues aspartate d glutamate e and lysine k are associated with high solubility whereas the aromatic residues phenylalanine f tryptophan w and tyrosine y are associated with low solubility fig c and supplementary fig sa  cysteine residue c has the lowest weight probably because disulfide bonds couldnt be properly formed in the e coli expression hosts stewart aslund and beckwith  rosano and ceccarelli  jia and jeon  aslund and beckwith   the weights are likely different if the solubility analysis was done using the reductasedeficient e coli origami host strains or eukaryotic hostshigher helix propensity has been reported to increase solubility idicula thomas and balaji  huang et al    however our analysis has shown that helical and turn propensities anticorrelate with solubility whereas sheet propensity lacks correlation with solubility suggesting that disordered regions may tend to be more soluble fig   in accordance with these swi has stronger negative correlations with helix and turn propensities these findings also suggest that protein solubility can be largely explained by overall amino acid composition not just the surface amino acid residues this idea aligns with our understanding that protein solubility and folding are closely linked and folding occurs cotranscriptionally a complex process that is driven various intrinsic and extrinsic factors wilkinson and harrison  chiti et al  tartaglia et al  diaz et al   however it is unclear why sheet propensity has little contribution to solubility because sheets have been shown to link closely with protein aggregation idiculathomas and balaji  we conclude that swi is a wellbalanced index that is derived from a simple sequence composition scoring method to demonstrate the usefulness of swi we developed a web server called sodope soluble domain for protein expression httpstisignercomsodope  sodope calculates the probability of solubility of a userselected region based on swi which can either be a fulllength or a partial sequence see methods and supplementary  table s  this implementation is based on our observation that some protein domains tend to be more soluble than the others to demonstrate this point we have analysed three commercial monoclonal antibodies and the severe acute respiratory syndrome coronavirus proteomes sarscov and sarscov wang et al  marra et al  f wu et al  supplementary fig s and s  these soluble domains may enhance protein solubility as a whole sodope also provides options for solubility prediction at the presence of solubility fusion tags similarly solubility tags may act as soluble protein domains that can outweigh the aggregation propensity of insoluble proteins however some soluble fusion proteins may become insoluble after proteolytic cleavage of solubility tags lebendiker and danieli   in addition sodope is integrated with tisigner a gene optimisation web service for protein expression this pipeline provides a holistic approach to improve the outcome of recombinant protein expression protein sequence properties the standard protein sequence properties were calculated using the biosequtilsprotparam module of biopython v cock et al    all miscellaneous protein sequence properties were computed using the r package protr v n xiao et al   protein solubility prediction we used the standard and miscellaneous protein sequence properties to predict the solubility of the psibiology and esol targets n and  respectively seiler et al  niwa et al    for method comparison we chose the protein solubility prediction tools that are scalable table   default configurations were used for running the command line toolsto benchmark the wall time of solubility prediction tools we selected  sequences that span a large range of lengths from the psibiology and esol datasets from  to  residues all the tools were run and timed using a single process without using gpus on a high performance computer [ usrbintime f e command  centos linux  core operating system  cores in  broadwell nodes ev  ghz dual socket  cores per socket  gib memory] single sequence fasta files were used as input files swi to improve protein solubility prediction we optimised the most recently published set of normalised bfactors using the psibiology dataset smith et al  fig   to avoid including homologous sequences in the test and training sets we clustered the psibiology targets using usearch v bit edgar   histag sequences were removed from all sequences before clustering to avoid false cluster inclusions we obtained  clusters using the parameters cluster_fast input_file id  msaout output_file threads   these clusters were divided into  subsets with approximately  sequences per subsets manually  the subsequent steps were done with histag sequences we used smith et al s normalised bfactors as the initial weights to maximise auc using these  subsets with a fold crossvalidation since auc is nondifferentiable we used the neldermead optimisation method implemented in scipy v which is a derivativefree heuristic simplexbased optimisation oliphant  millman and aivazis  nelder and mead   for each step in crossvalidation we used  bootstrap resamplings containing  soluble and  insoluble proteins optimisation was carried out for each sample giving  sets of weights the arithmetic mean of these weights was used to determine the training and test auc for the crossvalidation step fig a  bit score to examine the enrichment of amino acid residues in soluble and insoluble proteins we compute the bit scores for each amino acid residue in the psibiology soluble and insoluble groups  supplementary fig sa  we normalised the count of each residue in each x  group by the total number of residues in that group we used the normalised count of amino acid residues using the esol e coli sequences as the background the bit score of residue for soluble or insoluble group is then given by the following equationwhere is the normalised count of residue in the psibiology soluble or insoluble x f i x  group and is the normalised count in the esol sequences x f esol for a control random protein sequences were generated by incrementing the length of sequence starting from a length of  residues to  residues with a step size of  residues a hundred random sequences were generated for each length giving a total of  unique random sequences the sodope web server to estimate the probability of solubility using swi we fitted the following logistic regression to the psibiology dataset robability of solubility  exp ax b p      where is the swi of a given protein sequence and  the x  a    b   pvalue of loglikelihood ratio test was less than machine precision equation  can be used to predict the solubility of a protein sequence given that the protein is successfully expressed in e coli  supplementary table s fulllength sequence to examine the probability of solubility flexibility and gravy this functionality enables protein biochemists to plan their experiments and opt for the domains or regions with high probability of solubility furthermore we implemented a simulated annealing algorithm that maximised the probability of solubility for a given region by generating a list of regions with extended boundaries users can also predict the improvement in solubility by selecting a commonly used solubility tag or a custom tagwe linked sodope with tisigner which is our existing web server for maximising the accessibility of translation initiation sites bhandari lim and gardner   this pipeline allows users to predict and optimise both protein expression and solubility for a gene of interest the sodope web server is freely available at httpstisignercomsodope   code and data availability jupyter notebook of our analysis can be found at httpsgithubcomgardnerbinflabsodope_paper_  the source code for our solubility prediction server sodope can be found at httpsgithubcomgardnerbinflabtisignerreactjs                                                                                                         ccby  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi biorxiv preprint  ccby  international license authorfunder it is made available under a the copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg    ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n sarscov\n binding\n protein\n epitopes\n rbd\n ace\n proteins\n residues\n structure\n spike\n-----------------\nDOCUMENTS GROUP 5\n[' ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint concerns that sarscov will reemerge in the future consequently studies are underway to  determine the mechanisms through which sarscov interacts with the host immune system crosses  host barriers and other related aspects  sarscov was initially identified as having occurred through interspecies transmission from small  carnivores eg palm civets and raccoon dogs to humans in   in our current study we conducted a year followup study on a cohort of sarscovinfected  healthcare workers for analyzing the presence and persistence of antibodies against sarscov   ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint  the study site was a teaching hospital and the main center for treating clustered sars cases in is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint were missing in the followup table s respectively for failing to arrange a physical examination or due to resignation in addition serum  samples from another  subjects from the  sars cases were collected in  table s   another  nonsars infected healthcare workers from the same hospital were enrolled and serum  samples were collected as the control group to verify the end result data of the   sarscovinfected healthcare cohort in  table s     ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity which was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprintthe copyright holder for this preprint  httpsdoiorg doi medrxiv preprintthe copyright holder for this preprint  httpsdoiorg doi medrxiv preprintthe copyright holder for this preprint  httpsdoiorg doi medrxiv preprintwe found that the sarscovinfected healthcare workers who were treated with  corticosteroids such as prednisone and methylprednisolone exhibited lower igg titers during  the early stages of the disease than those who were not treated with corticosteroids although  not significantly for igg against n however no differences were found between the two  groups after  suggesting that corticosteroids may suppress the early production of  sarscov igg antibodies  boxplots and heatmaps were used to show the temporal trends and variations in the antibody levels  nonlinear exponential decay models were used to estimate the average decay rates of the antibody ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity    ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity  to analyze the general igg distributions subjects were divided into  groups group  n  including sarscovinfected healthcare workers enrolled in the original cohort n and  an additional group of sarscovinfected healthcare workers who were enrolled in   n and group  n of noninfected healthcare worker selected as the control the  igg against whole virus was above the cutoff value and the igg titers were significantly  higher in group  than in group  p   figure a  a similar trend was observed for  igg against n titers as group  had significantly higher igg titers than group  p  figure b  moreover in group  the positive rate of igg binding to whole virus and n  was  and  with statiscical significance p   respectively  kappa test was used to estimate the consistency between the methods for testing igg the table s    ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity  during the outbreak some sarscovinfected healthcare workers received corticosteroid  treatment while others did not the cohort was divided into two groups to assess the effect of antigen was used to detect its specific antibody which has been demonstrated to be specific in  clinical diagnosis and sarscov reservoir screening  we therefore adopted the testing of  igg against n combined with the titer analysis of igg against the whole virus to  comprehensively evaluate the sars antibody duration   ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity  in our cohort we found that two healthcare workers might have been misdiagnosed with  sars as the igg antibodies against both the whole virus and n antigen was persistently  lower than the cutoff value for these two patients notably all the cases enrolled in this study is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint   collectively based on our results we can infer that the igg against sarscov can persist at ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuity  ccbynd  international license it is made available under a is the authorfunder who has granted medrxiv a license to display the preprint in perpetuitywhich was not peerreviewed the copyright holder for this preprint  httpsdoiorg doi medrxiv preprint ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n medrxiv\n preprint\n license\n granted\n display\n perpetuityis\n ccbyncnd\n international\n made\n authorfunder\n-----------------\nDOCUMENTS GROUP 6\n['introduction pathogenic strain identification using lcmsmsbased proteomics presents a crucial yet highly challenging task many pathogenic strains feature significant phenotypic differences within a species with respect to pathogenicity zoonotic potential cell attachment and entry hostvirus interaction and clinical symptoms [] [] []  in a diagnostic context strainlevel knowledge is important to infer virulence   and drug resistance  for appropriate therapy however inferring exact strain information from proteomic samples remains a difficult task in particular when the taxonomic origin of a sample is unknown and when related strains feature high sequence similarity in recent years malditof mass spectrometry has gained popularity as fast sensitive and economical method for microbial biotyping however identifying strains using malditof workflows is still very challenging and requires curated often proprietary spectral databases   several commercial platforms for microbial biotyping down to the species or strain level are available based on maldi and other technologies such as the bruker maldi biotyper systems   the bruker strain typing with ir biotyper  and the ibis t universal biosensor   several studies report on limitations of malditof biotyping for strainlevel identifications and advocate advancements towards msms marker peptide detection and consequently the analysis was shifted to the msms level    in these studies however the msmsbased protein identification was established using sequence databases that were already targeted or restricted to particular species or limited sets in contrast untargeted msms typing approaches are limited to species level identification    however in general msms is preferred for the analysis of complex unpurified peptide mixtures as it is considered to provide more distinct and unambiguous peptide and protein identifications  and thus increased proteome resolution  as well as higher statistical confidence   in particular organisms with unknown taxonomic origin benefit from peptide sequencebased analysis as maldibased biotyping is in comparison too prone to ambiguous identifications   furthermore advances in instrumentation including higher resolution mass accuracy and dynamic range increasingly allow for identification of the majority of all fragmented peptides  resulting in higher sensitivity higher coverage of target proteomes and thus higher availability of distinctive features taking advantage of the vast amount of available protein sequences for msms strainlevel identification is challenging on the one hand constraining the search space may result in unidentified strains or incorrectly assigned taxa in particular for nonmodel organisms   on the other hand applying large databases is not recommended either since it decreases peptide identification rates  and thus eventually impedes taxonomic inference   furthermore with increasing database size sequence quality often decreases eg when using the complete ncbi protein in comparison to the ncbi refseq database and contaminant sequences may occur more often   therefore extended databases should only be used when necessary however strainlevel identification of msms spectra from samples with unclear taxonomic status requires an untargeted search against comprehensive databases holding as many strains as possible a common and popular concept to handle increased search spaces is the application of multiple identification steps in general independently of target application such as strainlevel identification these multistage identification search strategies are described by several different terms such as serial search   multistep   iterative    multistage   twostep [] [] [] as well as cascade search  and they find application in proteomics metaproteomics   and proteogenomics    most of these strategies do not only overlap in their objective of increasing the identification rate or identification confidence but share methodological principles as well this includes the concept of identifying primarily unassigned spectra using databases of increasing complexity for instance by employing altered digestion parameters additional posttranslational modifications or additional spectral and genomic databases [] [] [] []  as well as the recurring theme of database size reduction   [] [] []   in addition some methods rely on spectral quality assessment to enhance subsequent identification steps   or exhibit a focus on algorithmic runtime reduction   apart from database size multiproteome databases present an additional challenge for taxonomic assignment in the form of high sequence similarity between proteomes in particular between related species and strains these similarities give rise to taxonomically ambiguous widespread and false assignments and thus need to be accounted for several methods exist that provide various strategies to account for ambiguous peptide spectrum matches due to sequence similarity dworzanski et al apply a proteogenomic mapping approach in combination with discriminant analysis to infer most likely bacterial assignments   the proteogenomic approach was further developed in bacid that applies several statistical measures to account for similarity including the comparison of ratios of taxonomic differences and of unique peptides to known error rates ie the noise levels    bacid has since been applied to and extended for samples of unknown origin  as well as mixtures of bacteria    with the focus on metaproteomic analysis megan   unipept   metaproteomeanalyzer  and tcup  rely on lowest common ancestor or most specific taxonomy approaches to assign peptides spectrum matches to taxonomic levels micid uses a clustering approach based on peptidome similarity of taxa on different levels in combination with unified evalues to infer statistically significant representatives of clusters as identified or classified microorganisms    tracz et al use a direct assignment strategy by concatenating all proteins of a proteome into one pseudopolyprotein and considering only the topscored spectrum matches to a peptide as counts for bacterial candidates   in contrast pipasic explicitly makes use of ambiguous peptide spectrum matches psms by applying abundance correction based on intensityweighted proteome similarities of organisms in metaproteomic samples   while these methods share the objective of reducing uncertainty in taxonomic identification due to ambiguous assignments in the context of the protein inference issue   their application for untargeted strainlevel identification is not straightforward for several reasons for example unique peptides are not necessarily available for all candidate strains in the sequence databases for which the information given is often limited to species or higher taxonomic levels in addition the approaches rely on fully sequenced genomes and thus uniform proteome coverage or other forms of targeted curated inhouse or validated databases and are thus restricted to a predefined limited set of organisms often with definite focus on bacteria furthermore some methods do not scale well with immoderate large databases ie high number of proteomes with respect to computational or analytical performance previous multistep procedures illustrate the effect of database size on identification confidence and the advantage of applying concise prefiltered or specialized databases thus we transfer the general concept of multistep procedures to approach the increased search space necessary for untargeted and detailed taxonomic classification we present taxit an iterative workflow for untargeted strainlevel identification of protein samples by applying two separate identification steps for speciesand strainlevel classification we circumvent the immediate need for a comprehensive database containing strain all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint proteome sequences thereby a first untargeted search allows the selection of a relevant species and enables to focus a second search on a highly reduced but adequate choice of automatically downloaded strain proteomes resulting in increased identification confidence and reduced taxonomic ambiguity moreover strainlevel identification relies on recent data without the need of keeping a local comprehensive strain database uptodate the workflow takes advantage of a free publicly available and continuously growing protein sequence resource ncbi protein and is thus suitable for most common established ms experimental workflows material and methods the taxit workflow for strain identification consists of several recurring modules interconnected controlled in terms of input and output and automatically executed by the snakemake computational workflow management system   the designed workflow executes up to three stages including i an optional host filter ii species identification primary iteration and iii strain identification secondary iteration a concise overview of the workflow is illustrated in figure   the main steps of each iteration are the execution of a peptide search engine false discovery rate fdr control taxonomic classification taxa counting and adjustment as well as candidate selection and visualization the automated download of strain proteomes bridges the primary and secondary iteration the procedures of the main modules are described in detail below peptide search the central step in the analysis of msms spectra is an efficient peptide search for comprehensive protein identification therefore we rely on established and reliable opensource database search engines such as xtandem  in combination with the xtandem parser  and msgf   however any commandline search engine including proprietary ones could be implemented via additional snakemake rules we apply a classic targetdecoy approach for false discovery rate fdr control   decoy sequences are created upfront and independently of the search engine with fastadecoypl  by reversing the target sequences including contaminant proteins for instance crap   and both target and decoy databases are concatenated as suggested by jeong et al   psms are subjected to an fdr cutoff based on a permatch fdr calculated as ndecoyntarget with ndecoy being the number of decoys in between targets ntarget in a list of matches sorted by evalue    to acknowledge established false positive hits of previous iterations decoy sequences with an evalue above the fdrcutoff are passed on and concatenated to the database of the next iteration taxonomic classification we make use of the ncbi taxonomy to assign psms to corresponding taxa and redistribute shared hits introduced by proteins associated with higherupper taxonomic levels such as genus first ncbi protein accessions are mapped to ncbi taxids using the ncbi protein id mapping file ftpncbinihgovpubtaxonomyaccessiontaxidprotaccessiontaxidgz next taxonomic relations are inferred from the ncbi taxonomy nodes dump file ftpncbinihgovpubtaxonomytaxdumptargz and psms are reassigned to a taxonomic level based on the objective of the current iteration in general psms assigned to higher taxa such as genus are propagated as shared hits to corresponding leaf taxa as long as the target leaves already exhibit matches on their own in addition in the first iteration psms are summarized at the species level count adjustment for each candidate taxon ie species in primary and strains in secondary iteration respectively raw counts are calculated by summing over all assigned psms including nonunique matches to account for taxonomic biases caused by shared matches we integrate a weighting scheme based on the level of uniqueness using the global frequency of a psm here a psm count per taxon is adjusted by the number of occurrences in all candidate taxa thereby unique psms gain in value for the final taxa selection without fully neglecting the importance of high numbers of nonunique matches that are often highly present within closely related taxa such as strains selection and downloads after count adjustment the most dominant candidate taxon is selected as the most likely species or strain respectively in the final step of the first iteration the selected species is used to retrieve strainlevel data for the strain identification in the second iteration once more we rely on the ncbi taxonomy and infer all available strains for the candidate species via the nodes dump file next strain proteins are automatically downloaded from ncbi protein using the ncbi entrez api  in combination with the jsoup java html parser   this includes all available refseq as well as nonrefseq sequences since the availability of curated strain material is often limited finally the obtained protein sequences are merged into a single database and redundant entries are removed using seqkit rmdup   benchmarking experiments for benchmarking we compare taxit against classic comprehensive search strategies based on straight noniterative taxonomic identification supported by unique psms or abundance similarity correction as provided by pipasic   taxit uses ncbi refseq proteins of selected kingdoms as reference databases for initial species identification followed by automated and selective strain protein incorporation uniquepsmsand pipasicbased strategies however apply comprehensive databases integrating as many strainlevel sequences as possible at once including all protein sequences from the ncbi protein database for selected kingdoms in general a preselection of kingdoms may be justified by clinical findings based on for instance symptoms or microscopic examination   both uniquepsmsand pipasicbased strategies use the same procedures for peptide search fdr control and taxonomic classification as described in the iterative workflow however psms are not summarized at species level and counts are directly inferred at the lowest possible taxonomic level for the uniquepsmsbased strategy adjusted counts are based on psms that occur only once in other words only spectra assigned unambiguously to only one peptide sequence and thus organism are taken into account the abundance similarity correction of pipasic uses the similarity of expressed proteomes between taxa to account for attribution biases originally intended for metaproteomic abundance correction it is here applied to highlight the most likely strain within a provided sample all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint since pipasic is sensitive to a high number of taxa we limit the input to taxa with a minimum of two hits as well as to the  most abundant taxa expressed proteins per taxa are extracted according to the taxonomic classification and digested peptides with a minimum length of six amino acids are prepared using trypsin digestion   psms and tryptic peptides are then passed to pipasic to obtain corrected relative counts we perform all three strategies on several viral and bacterial msms spectra samples with available strainlevel knowledge this includes a cowpox virus brighton red strain acquired inhouse an avian infectious bronchitis virus strain beaudette ck  and a bacillus subtilis bsn  that we will refer to as cowpox bronchitis and bacillus sample respectively b subtilis bsn is a transgenic organism resulting from horizontal gene transfer hgt of the diviva protein from listeria monocytogenes strain egde to bacillus subtilis subsp subtilis str  since b subtilis bsn is neither yet present in the ncbi taxonomy nor in the ncbi protein database and only one protein is modified we expect b subtilis subsp subtilis str  to be selected as final strain candidate bacillus samples are examined twice once completely with  spectra bacillus all and once randomly reduced to  spectra bacillus k to improve performance with respect to the vast bacterial search space a detailed description of the cowpox sample acquisition and the search parameters for all samples is provided in supplementary item for the viral samples we used all viral ncbi refseq proteins  entries via entrez on july   in the iterative workflow and additionally all viral nonrefseq proteins  entries in total via entrez on july   for the uniquepsmsand pipasicbased strategies viral spectra were filtered beforehand using corresponding host proteomes all isoforms including the uniprot homo sapiens reference proteome  entries up may   for the cowpox sample and gallus gallus red jungle fowl reference proteome  entries up may   for the bronchitis sample for the iterative examination of the bacillus samples all bacterial ncbi refseq proteins  entries release  were downloaded via ftp ftpncbinlmnihgovrefseqreleasebacteria since downloading all ncbi protein entries for bacteria taxid  is impractical and requires too much time using the entrez api we used the ncbi blast nr database as most comprehensive common and readily available protein resource this database can be obtained in fasta format via ftp ftpncbinlmnihgovblastdbfasta release  and the bacteria subset  entries was extracted with inhouse scripts to create the databases for the uniquepsmsand pipasicbased strategies results to demonstrate the potential of iterative strainlevel identification we compared taxit against classic comprehensive search strategies based on noniterative taxonomic identification supported by either unique psms or the abundance similarity correction of pipasic   the final selections of the top taxa candidates for all samples and all three compared identification strategies are summarized in table   for the cowpox sample identification results agree the most taxit figure  and pipasic supplementary item   figure s  are both able to identify the expected cowpox virus brighton red strain however unique psms are limited to the parent cowpox virus species and not available at the strain level for this reason an incorrect identification of bat astrovirus hil gx bszt could occur based on a single unique psm supplementary item  figure s  the results also show that the original taxit counts are not an all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint appropriate measure to distinguish the expected strain from competing candidates figure   however the weightbased correction method implemented in taxit resolves the present tie and emphasizes the correct cowpox virus strain brighton redin comparison bronchitis and bacillus samples feature notable variability in proposed taxa candidates for bronchitis taxit could identify the expected avian infectious bronchitis virus strain beaudette ck strain despite applying a small correction based on weighting however final candidate selection is not limited to a single strain but additionally includes a closely related strain namely avian infectious bronchitis virus strain beaudette us  figure a  nevertheless taxit provides a highly constrained selection of candidates in the first place in contrast the pipasicbased strategy results in considerably more initial candidates and eventually promotes an incorrect strain namely avian infectious bronchitis virus strain  figure b  since no unique psms are available for the bronchitis sample strainlevel identification was not possible with this strategyusing taxit the expected b subtilis str  strain is identified correctly in the reduced as well as in the complete sample supplementary item   figure s and s  pipasic consistently misses the correct strain and species in favor of incorrect strains such as bacillus cereus sj and bacillus cereus b the uniquepsmsbased strategy includes the correct strain b subtilis str  into the final candidate list of the bacillus k sample however it fails to separate the strain from several distinct species due to equal amounts of unique psms supplementary item  figure s  furthermore the data analysis on the complete bacillus sample results in the incorrect species paenisporosarcina quisquiliarum being predominantly present in terms of uniquely assigned psmsbesides identification efficiency computational performance may delimit the potential application as routine or expeditious method in research or clinics therefore we compared runtime and memory consumption for all samples and strategies as illustrated in table  and figure   analysis was performed with xtandem as database search engine and limited to  threads on a server with debian gnulinux  jessie  cores  threads of type intelr xeonr cpu e v   ghz  gb of ram and ssd storage applying the iterative approach reduces memory usage down to one third for viral strain identification and two third for bacterial strain identification while the runtime of pipasic is comparably high taxit shows no substantial change in runtime in comparison to the uniquepsmsbased strategy for small databases such as the collective viral sequences or small sample sizes however analyzing the full bacillus sample reveals an increased runtime when using ncbi refseq proteins plus selected strain proteomes instead of the extensive ncbi blast nr database discussion taxit demonstrates superior performance with respect to strainlevel identification and computational expense in summary our approach can be used to unambiguously identify candidate organisms of all samples down to a low taxonomic level with the minor exception of a tie for the bronchitis sample in contrast the uniquepsmsbased identification strategy is repeatedly deficient at the strain level or features highly ambivalent results furthermore pipasic frequently favors an incorrect strain or even an all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint incorrect species for some samples correct strains are observed as a top candidate even in the original counts independently of database choice and prior to count adjustments nevertheless all samples benefit either from the iterative and focused database usage the count adjustment procedure or the reduced resource consumption while the outcome remains legitimate it was shown that taxit improves taxonomic assignment already in the countbased ranking independently of count adjustments by limiting candidates to strains of a single species early on  figure    for instance the search of bronchitis samples against the ncbi blast nr bacteria subspace results in a rather uniform distribution of original counts for infectious bronchitis virus strains the strains beaudette beaudette ck and beaudette uk are only slightly increased in comparison to other strains figure b  original  in contrast the iterative approach results in less strain candidates of the same species focusing solely on beaudette strains when selectively searching against avian coronavirus strains in the secondary iteration figure a original  this is a consequence of how and whether parental or multispecies proteins are associated with strains or taxa in general within the ncbi taxonomy in case of the iterative approach fewer mutual species or genus proteins are consulted in the strain identification iteration however the extent of this effect varies between samples candidate strains or target databases and taxonomies respectively for instance bacterial strain proteomes such as the bacillus subtilis strains feature numerous directly assigned mutual protein sequences and thus result in an extended range of candidate strains of the same species supplementary item  figure s and s  nevertheless the restriction to a specific set of strain proteomes prevents manifold primary misassignments to distant species genera or even phyla as can be observed for pipasicand uniquepsmsbased original counts respectively that cannot be sufficiently resolved after correction supplementary item   figure s and s  in general the iterative and selective database usage ensures that final strain selection is limited to strain candidates of an appropriate species thus it prevents false positive hits on distinct strains of other taxa including species genera and phyla and allows for a more confident final strain candidate selection furthermore uniform distributions and even consensus in original counts of strain candidates demonstrate the need and benefit of count adjustment methods the implemented weighting procedure can resolve ties between strains such as in the taxit cowpox sample analysis figure   or at least amplifies the correct strain and increases the distance to competing candidates taxit infers exactly one strain for the presented samples each expect for the bronchitis sample where the strains beaudette ck and beaudette uk cannot be differentiated we observed that the corresponding psms are fully shared between the two proteomes although different proteins are available for each strain in general peptide hits are either assigned to shared proteins of the parent strain infectious bronchitis virus or to homologous proteins that differ only in identifier but not in sequence though a more granular taxonomic relation cannot be ascertained from the ncbi taxonomy we expect the beaudette strains to feature a considerably closer relationship as compared to other infectious bronchitis virus strains therefore we consider the draw between beaudette strains as sufficiently appropriate strain identification for the uniquepsmsbased strategy we observed a poor availability of unique psms at the strain level while the exploitation of purely unique features is a common theme for specieslevel identification the low amount of unique psms in strains is insufficient for strainlevel inference however the frequency of all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint spectra matching to distinct proteins and proteomes remains a valuable parameter for strain differentiation when considering and weighing both unique and the plethora of nonunique matches taxit has a comparable computational runtime for small samples and databases such as the viral data despite using a constrained search space this is primarily a result of the additional strain proteome downloads since the ncbi entrez api is not designed and optimized for large scale downloads and proteins need to be fetched in numerous iterations of small chunks however the download overhead fades into the background when considering full bacterial samples such as bacillus all table  and figure   and gives place to a runtime improvement of three quarters when compared to ncbi blast nr database searches in contrast the runtime of pipasic is afflicted with additional sequence comparisons necessary for constructing the similarity matrix that is highly influenced by increasing numbers of psms and taxa to compare finally the memory footprint of taxit in comparison to the uniquepsmsand pipasicbased strategies remains constantly less for all samples as would be expected when using substantially less proteins in the search databases on a final note strainlevel identification performance is generally limited by the availability and integrity of taxa and proteomes in used databases however constantly increasing quality and quantity of the ncbi taxonomy and protein databases will induce constant improvement of strainlevel identification strategies such as the presented iterative workflow conclusion untargeted strain level identification via msms spectra is a challenging task with respect to the excessive quantity of strains that need to be considered competitively for solving this problem we present taxit an iterative approach that first focuses on species identification and thus limits strain identification to concise selected target databases in our method both iteration steps take advantage of publicly available data from the ncbi taxonomy and protein databases thereby taxit enables final strain identification from recent and relevant sequence data without the need of heavily curated tailored or taxonomically constrained databases taxit supports any msms peptide data suitable for classic database search methods with further ongoing developments our tool might enable a wide range of applications in public health research and potentially also clinics in the future based on further validation taxit might support diagnosis and treatment of pathogenbased infectious diseases taxit is available for download under opensource license at httpsgitlabcomrki_bioinformatics all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint figure   overview of the taxit workflow in up to three stages msms spectra are searched against a host proteome to prefilter host proteins optional against an untargeted database for global species identification primary iteration and against a targeted speciesbased and automatically fetched database for strain level identification secondary iteration the copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder the copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint rss resident set size tof timeofflight all rights reserved no reuse allowed without permissionthe copyright holder for this preprint which was not peerreviewed is the authorfunder  httpsdoiorg doi biorxiv preprint ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n preprint\n viruses\n species\n sequences\n biorxiv\n genome\n sequence\n al\n viral\n virus\n-----------------\nDOCUMENTS GROUP 7\n['introduction since december  an increasing number of pneumonia cases caused by a novel coronavirus covid have been identified in wuhan china   this new pathogen has exhibited high humantohuman transmissibility with approximately  confirmed cases and  deaths reported globally as of february  on january   wuhan a city in china of  million residents was forced to shut down both outbound and inbound traffic in an effort to contain the covid outbreak ahead of the lunar new year however it is estimated that more than five million people had already left the city before the lockdown   which has led to the rapid spread of covid within and beyond wuhanin addition to population mobility and humantohuman contact environmental factors can impact droplet transmission and survival of viruses eg influenza but have not yet been examined for this novel pathogen absolute humidity defined as the water content in ambient air has been found to be a strong environmental determinant of other viral transmissions    for example influenza viruses survive longer on surfaces or in droplets in cold and dry air increasing the likelihood of subsequent transmission thus it is key to understand the effects of environmental factors on the ongoing outbreak to support decisionmaking pertaining to disease control especially in locations where the risk of transmission may have been underestimated such as in humid and warmer locationsour contribution we examine variability in absolute humidity and transmission of covid across provinces in china and other select locations we show that the observed patterns of covid are not completely consistent with the hypothesis that high absolute humidity may limit the survival and transmission of this new virus data and methods epidemiological data to conduct our analysis we collected epidemiological data from the johns hopkins center for systems science and system website  incidence data were collected from various sources including the world health organization who us centers for disease control and prevention china center for disease control and prevention cdc european centre for disease control and prevention the chinese national health center nhc as well as dxy a chinese website that aggregates nhc and local ccdc situation reports in near realtime daily cumulative confirmed incidence data were collected for each province in china from january   ie the closure of wuhan to february   for comparison we also obtained epidemiological data for affected countries including thailand singapore japan and south korea as well as other regions in china with important differences in data collection  all rights reserved no reuse allowed without permissionauthorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi medrxiv preprint including hong kong and taiwanestimation of a proxy for the reproductive number based on the cumulative incidence data for each province we estimated a proxy for the reproductive number r in a collection of   and day intervals   r is a measure of potential disease transmissibility defined as the average number of people a case infects before it recovers or dies our proxy for r designated as r proxy  is a constant that maps cases occurring from time t to time t  d onto cases reported from time t  d to time t  d where d is an approximation of the serial interval ie the number of days between successive cases in a chain of disease transmission for multiple time points t we obtained values of r proxy t d given by where the values of d range from [ to ]  consistent with reported estimates of serial intervals  our measure is considered only a proxy for r because it does not use details of the so far poorly defined serial interval distribution but instead simply calculates the multiplicative increase in the number of incident cases over approximately one serial interval such proxies are at least approximately monotonically related to the true reproductive number and cross  when the true reproductive number crosses    after computing these proxy values over a variety of subsequent moving time windows a mean value was obtained and utilized as our estimated reproductive number r for each province to minimize potential inclusion of imported cases in our analysis r estimates were calculated after the closure of wuhan on january rd weather data we obtained monthly values of temperature and relative humidity data in january  for each provincial capital in china from world weather online   we assumed that the majority of disease incidence for each province would occur or be reported in or near the capital due to increased population density in these metropolitan areas we then calculated the absolute humidity from these two variables for each province using the following formula which is an approximation of the clausius clapeyron equation  where ah is the absolute humidity t is the temperature in degrees celsius rh is the relative humidity in percent  and e is the base of the natural logrelationship between local exponential growth and environmental factors we used two different interpolation approaches a loess regression and an exponential fit to visually identify the relationship between absolute humidity and our proxy for the reproductive number an indicator of the observed exponential growth rate for each location in order to identify the sta table   relationship between local exponential growth logr proxy  and environmental factors ie absolute humidity and temperaturetistical relevance of the relationship between the local reproductive number of covid cases and environmental factors we fitted a linear model using the logarithm of the local reproductive number r proxy as our response variable absolute humidity and temperature were included as independent variables results reproductive number proxy our estimates of r proxy  for each province within china and other countries appeared to be consistent across the range of serial intervals we analyzed as shown by the vertical lines displayed in figure  for each location most regions demonstrated r proxy estimates well above  signaling the likely presence of local exponential growthunivariate relationship with absolute humidity as shown by two separate data interpolation approaches in figure   observed patterns of transmissibility as a function of absolute humidity were mixed specifically panel a in figure  loess regression shows that not only dry and cold locations experience high values of r proxy as in influenza but locations with high absolute humidity may also have higher values of r proxy  while the magnitude of its variability is small and all values are well above relationship with environmental factors the regression model demonstrates that both absolute humidity and temperature are associated with local exponential growth of covid across provinces in china and other affected countries table   absolute humidity and temperature yielded a positive relationship and a slight negative relationship respectivelylimitations our estimates of the observed r proxy across locations were calculated using available and likely incomplete reported case count data with date of reporting rather than date of onset which adds noise to the estimation in addition the relatively short time length of the current outbreak combined with imperfect daily reporting practices make our results vulnerable to changes as more data becomes available we have assumed that travel limitations and other containment interventions have been implemented consistently across provinces and have had similar impacts thus population mixing and contact rates are assumed to be comparable and have ignored the fact that different places may have different reporting practices further improvements could incorporate data augmentation techniques that may be able to produce historical time series with likely estimates of case counts based on onset of disease rather than reporting dates this along with more detailed estimates of the serial interval distribution could yield more realistic estimates of r finally further experimental work needs to be conducted to better understand the mechanisms of transmission of the covid mechanistic understanding of transmission could lead to a coherent justification of our findings conclusion sustained transmission and rapid exponential growth of cases are possible over a range of humidity conditions ranging from cold and dry provinces in china such as jilin and heilongjiang to tropical locations such as guangxi and singapore our results suggest that changes in figure   absolute humidity in each provincial capital vs covid r proxy estimate calculated after the closure of wuhan city using data from  to  the size and color of each pin indicate cumulative cases per province and r proxy range respectivelyweather alone ie increase of temperature and humidity as spring and summer months arrive in the north hemisphere will not necessarily lead to declines in case counts without the implementation of extensive public health interventions further studies on the effects of absolute humidity and temperature on covid transmission are neededall rights reserved no reuse allowed without permissionauthorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi medrxiv preprint ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n cases\n medrxiv\n preprint\n wuhan\n number\n license\n china\n epidemic\n covid\n january\n-----------------\nDOCUMENTS GROUP 8\n['introduction  influenzalike illnesses ilis represent a significant burden on public health and can be caused by  a range of respiratory viruses in addition to influenza virus itself  an ongoing goal of antiviral drug  discovery is to develop broadlyacting therapeutics that can be used in the absence of definitive diagnosis  such as in the case of ilis for such strategies to succeed drug targets that are shared across virus families  need to be identified  for the spint inhibition assays trypsin which typically resides in the intestinal tract and expresses a very  broad activity towards different ha subtypes and hmpv f served as a control  in addition furin was  used as a negative control that is not inhibited by spint as none of the peptides used in combination  with the aforementioned proteases has a furin cleavage site we tested furinmediated cleavage on a  peptide with a hn hpai cleavage motif in the presence of  nm spint supplementary figure   we continued by measuring the vmax values for each proteasepeptide combination in the presence of  different spint concentrations and plotted the obtained vmax values against the spint concentrations  on a logarithmic scale supplementary figure   using prism software we then determined the ic that  reflects at which concentration the vmax of the respective reaction is inhibited by half spint cleavage inhibition of a representative hn cleavage site by trypsin results in an ic value of  nm table a   while the inhibition efficacy of spint towards matriptase hat klk and klk ranged from  nm to   nm table a  however inhibition was much less efficient for plasmin compared with trypsin   nm we observed a similar trend when testing peptides mimicking the hn and hn ha cleavage sites  using trypsin hat klk plasmin and trypsin matriptase plasmin respectively table a  with the  exception of plasmin we found that human respiratory tract proteases are inhibited with a higher efficacy  compared to trypsin we expanded our analysis to peptides mimicking ha cleavage sites of hn hn  lpai and hpai hn and hn that all reflected the results described above table a  only cleavage  inhibition of hn ha by klk did not significantly differ from the observation made with trypsin table     a  when we tested the inhibition of hmpv cleavage by trypsin matriptase and klk spint demonstrated  high inhibition efficacy for all three tested proteases with measured ics for trypsin matriptase and klk  of  nm  nm and  nm respectively table b compared to the ic values observed with  the peptides mimicking influenza ha cleavage site motifs the ic values for the hmpv f peptide were  very low   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint vivo situation and requires validation by expressing the fulllength fusion proteins in a cell culture model  to test cleavage and cleavage inhibition of the respective protease  however before conducting these  experiments we wanted to ensure that spint does not have a cytotoxic effect on cells therefore t  cells were incubated with various concentrations of spint over a time period of  hours pbs and m  ho served as cytotoxic negative and positive controls respectively we observed a slight reduction of  about    in cell viability when spint was added to the cells figure   to test spintmediated cleavage inhibition of fulllength ha we expressed the has of aca  hn ax hn and ashanghai hn in t cells and added recombinant matriptase or  klk protease that were preincubated with nm or nm spint trypsin and the respective protease  without spint incubation were used as controls cleavage of ha was analyzed via western blot and the  signal intensities of the ha bands were quantified using the control sample without spint incubation as  a reference point to illustrate the relative cleavage of ha with and without inhibitor figure a and bd   trypsin cleaved all tested ha proteins with very high efficiency that was not observed with matriptase or  klk  figure b d however hn ha was cleaved by matriptase and klk to a similar extent without  and with nm spint nm spint led to a relative cleavage reduction of about  and  for  matriptase and klk respectively figure a and b klkmediated cleavage of hn ha was reduced by  about  when klk was preincubated with nm spint and by about  when nm spint was  used figure a and c when we tested the cleavage inhibition of matriptase with hn ha as a substrate  we found that nm and nm spint reduced the cleavage to  and  cleavage respectively  compared to the control figure a and d in contrast nm spint had no effect on klkmediated  cleavage of hn ha while nm reduced relative cleavage by approximately  figure a and d    ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  https doiorg doi biorxiv preprint in order to determine whether spint also prevented cleavage of hmpv f we first examined which  proteases in addition to trypsin and tmprss were able to cleave hmpv f first we cotransfected the  full length tmprss hat and matriptase with hmpv f in vero cells the f protein was then radioactively  labeled with  s methionine and cleavage was examined by quantifying the f full length protein and the  f cleavage product we found that tmprss and hat were able to efficiently cleave hmpv f while  matriptase decreased the expression of f though it is not clear if this was due to general degradation of  protein or lower initial expression however matriptase demonstrated potential lowlevel cleavage when  cotransfected  figure a and b we then examined cleavage by the exogenous proteases klk klk  and matriptase compared with the trypsin control klk and matriptase were able to cleave hmpv f while klk was not  figure c and d in agreement with the peptide assay cleavage of hmpv f by klk  and matriptase was less efficient than for trypsin and both peptide and fulllength protein assays  demonstrate that klk does not cleave hmpv f this also serves as confirmation that matriptase likely  cleaves hmpv f but coexpression with matriptase may alter protein synthesis stability or turnover if co expressed during synthesis and transport to the cell surface next we tested spint inhibition of  exogenous proteases trypsin klk and matriptase we preincubated spint with each protease added  it to vero cells expressing hmpv f and analyzed cleavage product formation spint preincubation  minimally affected cleavage at a concentration of nm but addition of nm spint resulted in  inhibition of trypsin klk and matriptasemediated cleavage of hmpv similar to our findings for ha   figure e and f  spint inhibits ha mediated cellcell fusion  the above presented biochemical experiments demonstrate that spint is able to efficiently  inhibit proteolytic cleavage of hmpv f and several influenza a ha subtypes by a variety of proteases for  a functional analysis to determine whether spint inhibition prevents cells to cell fusion and viral growth  we examined influenza a infection in cell culture while hmpv is an important human pathogen influenza   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint grows significantly better in cell culture compared with hmpv first we tested whether cleavage  inhibition by spint resulted in the inhibition of cellcell fusion as described above matriptase and klk  were preincubated with nm and nm spint and subsequently added to vero cells expressing  aca hn ha or ashanghai hn ha cells were then briefly exposed to a low ph  buffer to induce fusion and subsequently analyzed using an immune fluorescence assay when matriptase  and klk were tested with nm spint and incubated with vero cells expressing hn ha we still  observed syncytia formation  figure a  however mm spint resulted in the abrogation of syncytia  formation triggered by cleavage of the respective ha by matriptase and klk we made the same  observation when we tested klk and hn ha  figure b  matriptasemediated hn ha syncytia  formation was inhibited by the addition of nm spint  figure b  to ensure that cellcell fusion  inhibition is a result of ha cleavage inhibition through spint but not a side effect of spint treatment  per se we expressed avietnam hn ha in vero cells and treated them with the inhibitor  hn ha possesses a hpai cleavage site and is cleaved intracellularly by furin during its maturation  process spint does not inhibit furin and is not able to cross cell membranes thus spint can not  interfere with the proteolytic processing of hn ha and therefore this control allows to examine whether  spint interferes with cellcell fusion figure c shows that hn ha forms large syncytia in the absence  of spint as well as in the presence of  nm spint hence we conclude that spint does not have a  direct inhibitory effect on cellcell fusion  spint reduces viral growth in cell culture  to understand whether spint was able to inhibit or reduce the growth of virus in a cell culture  model over the course of  hours we transfected cells with human tmprss and human matriptase two  major proteases that have been shown to be responsible for the activation of distinct influenza a subtype viruses tmprss is essential for hn virus propagation in mice and plays a major role in the activation  of hn and hn viruses   matriptase cleaves hn ha in a subtype specific manner is involved   ccbyncnd  international license is made available under a in the in vivo cleavage of hn ha and our results described above suggest a role for matriptase in the  activation of hn    at  hours post transfection we infected mdck cells with aca hn  at a moi of  and subsequently added spint protein at different concentrations nontransfected cells  served as a control and exogenous trypsin was added to facilitate viral propagation the supernatants  were harvested  hours post infection and viral titers were subsequently analyzed using an immuno plaque assay spint initially mitigated trypsinmediated growth of hn at a concentration of nm and  the extent of inhibition slightly increased with higher concentrations  figure a  table a  the highest  tested spint concentration of nm reduced viral growth by about  log  figure a  table a  we  observed a similar pattern with cells transfected with human matriptase  figure b  table a  growth  inhibition started at a spint concentration of nm and with the application of nm growth was  reduced by approximately  logs  figure b  table a  when we infected cells expressing tmprss with  hn and added spint viral growth was significantly reduced at a concentration of nm addition of  nm spint led to a reduction of viral growth of about  logs  figure c  table a  we also tested  whether spint could reduce the growth of a hn virus because it is major circulating seasonal influenza  subtype however tmprss and matriptase do not seem to activate hn viruses    hence trypsin  and spint were added to the growth medium of cells infected with ax hn compared to control  cells without added inhibitor spint significantly inhibited trypsinmediated hn growth at a  concentration of nm  figure d  table a  at the highest spint concentration of nm viral growth  was reduced by about  log  figure d  we also examined the effect of spint inhibition of hmpv spread over time vero cells were  infected with rghmpv at moi  and subsequently treated with nm of spint and gml of tpck trypsin every  hours cells were scraped and the amount of virus present was titered up to hpi with  spint and trypsin replenished daily we find that untreated cells are infected and demonstrate  significant spread through hpi conversely cells infected and treated with spint had no detectible   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint virus up to hpi and very minimal virus detected at  and hpi demonstrating that spint significantly  inhibition hmpv viral replication and spread  figure    antiviral therapies are often applied when patients already show signs of disease therefore we  tested if spint was able to reduce viral growth when added to cells  hours after the initial infection cells were infected with  moi of aca hn and trypsin was added to promote viral growth at the time of infection we also added nm spint to one sample a second sample received nm  spint  hours post infection growth supernatants were harvested  hours later and viral growth was  analyzed we found that viral growth was significantly reduced by regardless whether spint was added  at the time of infection or  hours later  figure   table b   discussion  influenza a virus has caused four pandemics since the early  th century and infects millions of people each year as seasonal flu resulting in up to  deaths annually  vaccination efforts have  proven to be challenging due to the antigenic drift of the virus and emerging resistance phenotypes  moreover the efficacy of vaccines seems to be significantly reduced in certain highrisk groups   prevalent antiviral therapies to treat influenza a virusinfected patients such as adamantanes and  neuraminidase inhibitors target viral proteins but there is increasing number of reports about circulating  influenza a subtypes that are resistant to these treatments  hmpv causes infections in the upper and  lower respiratory tract expressing very similar symptoms as influenza infections and resulting in significant  morbidity and mortality    the most susceptible groups are young children older adults and people  that are immunocompromised  currently there are not treatment against hmpv infections  available in this study we focused on a novel approach that uses antiviral therapies targeting host factors  rather than viral proteins offering a more broad and potentially more effective therapeutic approach  we demonstrate that spint a potent inhibitor of serinetype proteases is able to significantly inhibit   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint cleavage of hmpv f and ha impair hatriggered fusion of cells and hence reduce the growth of various  influenza a strains in cell culture we assessed cleavage of the hmpv fusion protein in vitro using a peptide cleavage assay modified from  previously work on other viral fusion proteins   the hmpv f peptide was cleaved by trypsin  plasmin matriptase and klk but was unable to be cleaved by klk to confirm these findings in a  system in which the entire hmpv f protein was subject to cleavage we coexpressed the fusion protein  with tmprss hat and matriptase proteases or treated f with exogenous proteases klk klk and  matriptase these findings are the first to identify proteases besides trypsin and tmprss that are able  to cleave hmpv f in addition hmpv appears to utilize many of the serine proteases that influenza uses for ha processing and therefore offers strong potential for an antiviral target  spint demonstrates greater advantage over other inhibitors of host proteases such as eg aprotinin that  was shown to be an effective antiviral but also seemed to be specific only for a subset of proteases  it can be argued that a more specific protease inhibitor which inhibits only one or very few proteases  might be more advantageous because it may result in less side effects with respect to influenza a  infections tmprss could represent such a specific target as it was shown to a major activating proteases  for hn and hn in mice and human airway cells       however there is no evidence that  that application of a broadspectrum protease inhibitor results in more severe side effects than a specific  one as side effects may not be a consequence of the protease inhibition but the compound itself may act  against different targets in the body the reports demonstrating that tmprss is crucial for hn and  hn virus propagation in mice and cell culture suggest that it also plays a major role in the human  respiratory tract so far however it is unclear whether the obtained results translate to humans and other  studies have shown that for example human matriptase is able to process hn and hn   our peptide assay suggests that spint has a wide variety of host protease specificity with the exception of plasmin all the tested proteases in combination with peptides mimicking the cleavage site of different   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint  cleavage inhibition of hmpv f were substantially lower in the picomolar range this suggests that the  hmpv cleavage may be more selectively inhibited by spint however the western blot data showed  that addition of the lowest concentration  nm of spint did not result in cleavage inhibition of hmpv  f by the tested proteases differences in sensitivity of spint between influenza ha and hmpv require  further investigation  spint poses several potential advantages over other inhibitors that target host proteases cell culture  studies showed that for example matriptasemediated hn ha cleavage was efficiently inhibited at a  concentration of nm spint in contrast the substrate range for aprotinin a serine protease inhibitor shown to reduce influenza a infections by targeting host proteases seemed to be more limited  other  synthetic and peptidelike molecules designed to inhibit very specific serine proteases such as tmprss  tmprss and tmprssd hat were only tested with those proteases and their potential to inhibit other  proteases relevant for influenza a activation remains unclear  currently the most promising  antiviral protein inhibitor is camostat which is already approved in japan for the treatment of chronic  pancreatitis  recently it was demonstrated that camostat inhibited influenza replication in cell  culture and prevented the viral spread and pathogenesis of sarscov in mice by inhibition of serine  proteases   however camostat was applied prior to the virus infection and it was administered  into the mice via oral gavage   a previous study showed that spint significantly attenuated influenza  a infections in mice using a concentration that was x lower than the described camostat concentration  and intranasal administration was sufficient  our current study suggests that spint is able to  significantly inhibit viral spread during an ongoing infection and does not need to be applied prior or at  the start of an infection the mouse study also showed that spint can be applied directly to the  respiratory tract while camostat that is currently distributed as a pill and therefore less organ specific in  addition camostat is synthetic whereas spint is a naturally occurring molecule which may attenuate   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint potential adverse effects due to nonnative compounds activating the immune system future research  will be conducted to test if spint can be applied more efficiently via an inhaler and to explore potential  side effects in mice studies however when we tested the potential of spint to inhibit viral replication  in a cell culture model we were only able to achieve growth reductions of approximately   logs after   hours with a concentration of nm spint one potential explanation is that nm spint was  unable to saturate the proteases present in the individual experiments and was not sufficient to prevent  viral growth in addition the continuous overexpression of matriptase and tmprss may have produced  an artificially high quantity of protein that exceeded the inhibitory capacities of spint this problem  could be solved either by using higher concentration of spint or by optimizing its inhibitory properties however the data also demonstrates that spint has the ability to inhibit proteases that expressed on  the cell surface and that inhibition is not limited to proteases that were added exogenously and pre incubated with the inhibitor  figure   spint did not express any cytotoxic effects up to a concentration  of mm significantly above the therapeutic dosage required for inhibition in comparison with other  studies the spint concentration we used here were in the nanomolar range while other published  inhibitors require micromolar concentrations  however we believe that future research will allow  to fully utilize the potential of spint as a broadspectrum antiviral therapy wu et al recently described  that the kunitz domain i of spint is responsible for the inhibition of matriptase  in future studies we  will explore whether the inhibitory capabilities of spint can be condensed into small peptides that may  improve its efficacy its ability to inhibit a broad range of serine protease that are involved in the activation  of influenza a suggest that a spint based antiviral therapy could be efficient against other pathogens  too tmprss for example not only plays a major role in the pathogenesis of hn but is also required  for the activation of sarscov and merscov and hmpv   currently treatment options for these  viruses are very limited and therefore spint could become a viable option if its potential as an antiviral  therapeutic can be fully exploited   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint however while spint has a therapeutic potential to treat ilis caused by viruses that require activation  by trypsinlike serine proteases it may have its limitation to provide a treatment option for infections  caused by influenza hpai viruses such as hn  these viruses are believed to be activated by furin  and proprotein convertases that belong to the class of subtilisinlike proteases  preliminary data  from our lab demonstrated that spint did not inhibit furinmediated cleavage of hpai cleavage site  peptide mimics as well as peptides carrying described furin cleavage sites data not shown in addition  furin acts intracellularly and we have no evidence that spint is able to penetrate the cell membrane and  thus inhibiting proteases located in intracellular compartments therefore it seems unlikely that spint  is able to inhibit furin in cell culturebased studies or in vivo experiments in conclusion we believe spint has potential to be developed into a novel antiviral therapy in contrast  to most similar drugs that are synthetic spint is an endogenously expressed protein product that  confers resistance to a variety of pathogenic viruses and which can potentially be delivered directly into  the respiratory tract as an aerosol most importantly spint demonstrated the ability to significantly  attenuate an ongoing viral infection in cell culture and further research will be conducted to explore the  time period during which spint demonstrates the highest efficacy  materials and methods  cells plasmids viruses and proteins  t vero and mdck cells american type culture collection used for influenza experiments  were maintained in dulbeccos modified eagle medium dmem supplemented with  mm hepes  cellgro and  fetal bovine serum vwr vero cells used for hmpv experiments were maintained in  dmem hyclone supplemented with  fbs sigma the plasmid encoding aca hn ha was  generated as described   the plasmid encoding for hmpv f sh was generated as described the plasmids encoding for ashanghai hn ha human tmprss and human matriptase were  purchased from sino biological inc the plasmid encoding for aaichi hn ha was generously   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint donated by david steinhauer aca and ax viruses were propagated in eggs all recombinant  proteases were purchased as described  expression and purification of spint  spint was expressed and purified as described with minor modification  in brief e coli ril  de arctic express cells agilent were transformed with spintpsumo cells were then grown in  l luria broth containing  gml kanamycin at c at od  cells were chilled on ice and protein  expression was induced with  mm iptg cells were then grown over night at c cells were harvested  and protein was purified as previously described  spint protein was eluted by a hour incubation  with ulpxhis glycerol was added to the eluted spint protein to a final concentration of  and  protein aliquots were stored at c protein concentration was determined by analyzing different  dilutions of spint on an sdspage gel along with  defined concentrations of bsa between  ng and   g the gel was then stained with coomassie scanned with chemidoc imaging system biorad and  bands were quantified using image lab software biorad concentrations of the spint dilutions were  determined based on the bsa concentrations and the final spint concentration was calculated based on  the average of the spint different dilutions  peptide assays  peptide assays were carried out as described  the sequence of the hmpv f peptide mimicking  the hmpv f cleavage site used in this assay is enprqsrfvl including the same nand cterminal  modifications as described for the ha peptides  the vmax was calculated by graphing each replicate in microsoft excel and determining the slope of the reaction for every concentration  nm  nm  nm   nm  nm  nm  nm  nm  nm and  nm the vmax values were then plotted in the  graphpad prism  software against the log of the spint concentration to produce a negative sigmoidal  graph from which the ic or the concentration of spint at which the vmax is inhibited by half could be   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint extrapolated for each peptide protease mixture since the xaxis was the log of the spint  concentration the inverse log was then taken for each number to calculate the ic in nm  cytotoxicity assay  the cytotoxicity assay was performed with a cell counting kit dojindo molecular technologies  according to the manufacturers instructions in brief approx  x   t cells were seeded per well of  a well plate and grown over night spint was added at the indicated concentrations dmem and   m ho were used as a control  hours later  l ckk solution were added to each well and  incubated for  hour absorbance at  nm was measured using a spark microplate reader tecan per  sample and treatment three technical replicates were used and the average was counted as one biological replicate experiment was conducted three times  metabolic protein labeling and immunoprecipitation of hmpv fusion protein  vero cells were transfected with ug of pdna using lipofectamine and plus reagent invitrogen  in optimem gibco according to the manufacturers protocol the following day cells were washed with  pbs and starved in cysteinemethionine deficient media for  min and radiolabeled with ucml s cysteinemethionine for  hours cells were lysed in ripa lysis buffer and processed as described  previously  and the fusion protein of hmpv was immunoprecipitated using antihmpv f g  monoclonal antibody john williams u pitt samples were run on a  sdspage and visualized using  the typhoon imaging system band densitometry was conducted using imagequant software ge repository respective secondary antibodies had an alexa tag invitrogen western blot membranes  were scanned using a chemidoc imaging system biorad for quantification the pixel intensity of the  individual ha bands was measured using imagej software and cleavage efficiencies were calculated by  the following equation ha  nm or  nm spintha nm spint x  cellcell fusion assays  were carried out as described   inhibition of influenza viral infection in cell culture  mdck cells were seeded to a confluency of about  in well plates one plate each was then  transformed with a plasmid allowing for the expression of human matriptase or human tmprss one  plate was transformed with empty vector  hours post transfection cells were infected with the  respective egggrown virus at a moi of approx  different spint concentrations were added as  indicated  nm trypsin was added to the cells transformed with the empty vector  hours post  infection supernatants were collected centrifuged and stored at c viral titers were determined using  an immuneplaque assay as described   inhibition of hmpv viral infection in cell culture  vero  cells were plated into well plates the following day cells were infected with  moi  rghmpv for  hours cells were washed with pbs and l optimem with or without nm  spint and gml of tpcktrypsin was added and incubated for up to  hours the spint and trypsin  was replenished in new optimem every  hours for each time point media was aspirated and l  of optimem was added to cells followed by scraping and flash freezing these samples were then titered  on confluent vero cells up to   dilution to calculate viral titer graph shows  independent replicates  with internal duplicates plotted as individual points some data points not shown are due to sample loss  with a minimum of  points per group or  independent replicates   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it hepatocyte growth factor activator inhibitor placental bikunin haipb gene is frequently  hypermethylated in human hepatocellular carcinoma cancer res  the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint analysis was performed using a nonpaired students ttest comparing the samples tested with  nm  spint against the respective sample incubated with  nm spint error bars indicate standard  deviation  indicates p     figure   tmprss hat matriptase and klk cleave hmpv f and spint is able to prevent cleavage by  exogenous proteases hmpv f was either expressed alone or cotransfected with protease and allowed  to express for  hours cells were then metabolically starved of cysteine and methionine followed by  radioactive s labeling of protein for  hours in the presence of tpcktrypsin or specified protease  spint treated proteases were incubated at room temperature for  minutes and placed onto cells for   hours radioactive gels were quantified using imagequant software with percent cleavage equal  to        a and b cotransfected proteases tmprss hat and matriptase are able to cleave  hmpv f n while c and d exogenous proteases klk and matriptase but not klk are able to cleave  hmpv f n e and f spint prevented cleavage of hmpv f by trypsin klk and matriptase at nm  concentrations demonstrated by the loss of the f cleavage product n statistical analysis was  performed using a oneway anova followed by a students ttest with a bonferroni multiple comparisons  test correction p  p  p  p  n values represent independent  replicates for each treatment group error bars represent sd  the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint antibodies and a secondary fluorogenic alexa antibody nuclei were stained using dapi magnification  x c vero cells expressed avietnam hn ha that was cleaved during its maturation  process in the cell spint was added at  nm or  nm at the time of transfection magnification x  for human matriptase or human tmprss and allowed to express the proteins for  hours cells  expressing human matriptase b or human tmprss c were then infected with aca hn at a  moi of  and different spint concentration were added to each well nontransfected cells to which  trypsin was added served as a control a d mdck cells were infected with ax hn at an moi of   and trypsin was added to assist viral propagation different spint concentration were added as  indicated after  hours of infection the supernatants were collected and used for an immunoplaque  assay to determine the viral loads experiment was repeated three times and each dot represents the viral  titer of a single experiment statistical analysis was performed using a nonpaired students ttest  comparing the control  h against the respective sample error bars indicate standard deviation   indicates p    extended horizontal line within the error bars represents mean value of the three  independent replicates  completed in duplicate all data points plotted within the graph statistical analysis was performed using  a students ttest p   p   p   p   nd indicates that the sample was  below the limit of detection   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint   ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint     ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint  ccbyncnd  international license is made available under a the copyright holder for this preprint which was not peerreviewed is the authorfunder it  httpsdoiorg doi biorxiv preprint ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n cells\n preprint\n fig\n biorxiv\n expression\n cell\n ace\n mice\n copyright\n holder\n-----------------\nDOCUMENTS GROUP 9\n['introduction the model to represent the viral load decrease thereof  in this manuscript we first present the effect of isolation considering both home quarantine for  individuals that are part of a contact trace network and for infected individuals with mild symptoms  and hospital isolation for severe cases we argue that when an individual is quarantined at home  this will only result in a partial reduction of contacts accounting for contacts with household  members and other isolation imperfectness to compensate for this imperfect isolation we consider  the use of an antiviral compound we test these different control measures in a simulation study  that aims at representing given the available information the current covid epidemic while  at this phase of the epidemic many countries are already beyond the point of local containment we  do however expect that the methodology we propose will be key to avoid a second peak especially  given the limited depletion of susceptibles individuals are initially susceptible s and once infected they enter the exposed class e the on the notion of infectious contact processes  first contacts between individuals are generated  when such contacts are generated between susceptible and infectious people these can result in  an infection event according to a bernoulli probability value based on the time since infection  this probability is computed at a precise time point as the product of two components the  infectiousness measure t which quantifies the level of infectiousness over time and the total  amount of infectivity q ie the number of infections over the contact rate  the function t is  defined over the exposed and infectious period or analogously over the incubation and symptomatic  period along which it integrates to one this function is scaled to have a similar shape among  different infectives based on their lengths of exposed and infectious period according to this  framework an infectious individual makes effective contacts at a rate rt given bywhere  is the contact rate the mean number of effective contacts is an approximation of the  reproduction number the two quantities are identical in an infinite and homogeneous population  where the probability of making two effective contacts with the same person is zero for the  considered population size the probability of this event is extremely low therefore throughout  the manuscript we approximate the reproduction number with the mean number of secondary  cases  in the considered framework isolationquarantine is implemented by reducing the contact rate  at  the time of diagnosis to date little is known about the difference in viral load among severe and  mild cases zhou et al  indicate that in nasal and throat swabs the viral load is higher in mild cases  due to this uncertainty we assume that the same curve is defined for all the infected individuals  in this work we assume the population to be homogeneous closed and finite population these  assumptions relate to the control measures currently in place eg in italy aiming at containing  immigration and emigration in a country with an ongoing outbreak   simulation parameters and distributions in table  we report the parameters and distributions that were utilized in the simulation study  where distributions are not reported the parameters are assumed to be constant in the last  column when available we report the references to the literature that justifies the choice of the  parameter value or distribution we use  we assume infectious individuals to make on average between two and three effective contacts nasal and throat swabs data  the time to hospitalization is estimated from the data presented in  backer et al  we assume that the time to hospitalization coincides with the time of diagnosis at with sarscov a contact tracing procedure is started we assume that each contact in h i will  be traced back successfully with probability  depending on the considered scenario tracedback  individuals will be monitored or tested when individuals are found positive for sarscov they  will be put in quarantineisolation for certain scenarios next to quarantine we also inject infected  individuals with an antiviral drug ie remdesivir  we assume that the test can detect infection after two days since infection traced individuals who  test negative the first time are tested again after two days the quarantine will result in a decreased  contact rate ie imperfect isolation  q  while in case of perfect isolation the contact rate is set  to zero similarly diagnosed individuals will also be quarantined at home mild symptoms with  a decreased contact rate  q  or in the hospital severe symptoms where we assume that perfect  antiviral compounds to compensate for imperfect isolation we investigate the use of antiviral compounds to reduce the  infectiousness of an infected individual we assume that once the antiviral compound has been  administered the infectiousness measure will exponentially decay according to an inverse malthusian  growth model shown in figure    the rate of this decay is set to represent the reduction in viral  load due to remdesevir as reported in  for the mers coronavirus   all rights reserved no reuse allowed without permissionauthorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi medrxiv preprint figure   reduction of infectiousness the blue and the orange lines describe the infectiousness measure respectively before dashed blue and after antiviral injection solid yellow the red arrows indicate the injection times scenarios we assume the following parameters for the reduction of contacts because of home quarantine  q      and for the probability of tracing back a contact in history h i        scenarios in all the considered scenarios we assume that individuals are isolated or quarantined  when diagnosed moreover we assume that contact tracing starts at the time of diagnosis  ias traced individuals are monitored for two weeks and isolatedquarantined if they show symptoms during this period this scenario is similar to the baseline scenario described by hellewell et al  with the exception that in our description only severe cases are isolated while the mild are home quarantined this scenario reflects more realistically the current practice of containmentibs traced individuals are isolatedquarantined as soon as they test positive for sarscovwe assume that an individual that is infected tests positive  days after infection therefore a traced individual is tested immediately when traced and if this test was negative we test the individual again two days lateribtbs a diagnosed patient is immediately treated with the antiviral drug furthermore traced individuals are isolatedquarantined and injected with the antiviral drug as soon as they test positive for sarscov we assume that an individual that is infected tests positive  days after infection therefore a traced individual is tested immediately when traced and if this test was negative we test the individual again two days laterfor each scenario we run  simulations among these we compute the final size and the cases  at peak for the one in which at least the  of individuals have been infected doing this we only  account for outbreaks that are most challenging to contain authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the  httpsdoiorg doi medrxiv preprint to successfully trace contacts and on the reduction in contact rate due to quarantine figure  isolation and quarantine lead to a substantial decrease in final size and peak incidence when  performed prior to symptom onset their efficacy increases which is important as recent work  indicates that infectees are infectious prior to symptom onset  the antiviral treatment is shown for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q  together with the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks right panel mean peak incidence together with the   and the   percentiles  to have a substantial impact and together with quarantine and isolation significantly reduces the  final size the peak incidence and the number of outbreaks that are most challenging to contain discussion we assume that we have sufficient antiviral drugs doses to treat all individuals that are encountered  via the contact tracing procedure this is motivated by the fact that we consider an emerging  outbreak and the required number of doses will thus be limited  furthermore we assume that all the individuals will show symptoms sooner or later during their  infectious period and therefore all infected individuals will be diagnosed due to the awareness of show that for the tested scenarios the use of an antiviral drug increases the control of the outbreak  even when undiagnosed individuals are circulating in the population  in the hospital we assume perfect isolation meaning that infected individuals cannot spread the authorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the  conclusion the ongoing epidemic of covid threatens the health system of many countries although we consider a setting where the reproductive number between symptomatic and asymptomatic is  equal  figure   and a setting where the asymptomatic infectees have a reproduction number that is   of the reproduction number set for symptomatic infectees figure    for both settings we  show a significant advantage with respect to the reduction of the final size and the peak incidence  yet our analysis also shows that when the asymptomatic are as infectious as the symptomatic  infectees it is important to have a high contact tracing success rate   we remain hopeful that the ongoing clinical trials will reveal an antiviral compound that can be  appendix sensitivity analyses we report in figure  and  the sensitivity analysis for the quarantine contact rate  q     the introduction of an antiviral compound substantially contributes in reducing the final size the  peak incidence and the probability of a challenging outbreak in all the considered settings this  decrease compared to the scenario in which only isolationquarantine is implemented increases  when quarantine is less effective left panels in figure  and  we vary the reproduction number  that is set respectively to r    and r    the effect of the antiviral drug in addition to  isolation and quarantine increases when the reproduction number increases in case of r    case of remarkable impact in both the final size and the peak incidence  we tested also the effect of an asymptomatic proportion of infectives we assume that the  authorfunder who has granted medrxiv a license to display the preprint in perpetuity figure   mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q  left panel and  q  right panel together with  and  percentiles figure   final size distribution left panel and mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q  and r    in the left panel for each scenario we report the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks in the right panel together with the point estimates we report the  and   percentiles figure   final size distribution left panel and mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q  and r    in the left panel for each scenario we report the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks in the right panel together with the point estimates we report the  and   percentiles  all rights reserved no reuse allowed without permissionauthorfunder who has granted medrxiv a license to display the preprint in perpetuity figure   final size distribution left panel and mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q   r   and the test detect positively an infectious individual after  days since infection in the left panel for each scenario we report the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks in the right panel together with the point estimates we report the  and   percentilesnumber of value r    while in figure  the reproduction number for symptomatic and  asymptomatic is set respectively to r s   and r a   r s    also in these scenarios we noticed  that the administration of an antiviral drug lead to a substantial decrease in the final size and peak  incidence compared to the values obtained when the other control measures are considered   all rights reserved no reuse allowed without permissionauthorfunder who has granted medrxiv a license to display the preprint in perpetuitythe copyright holder for this preprint which was not peerreviewed is the figure   final size distribution left panel and mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q   and the proportion of asymptomatic individuals is  we consider the same reproduction number for asymptomatic r a   and symptomatic r s   individuals r a   r s    in the left panel for each scenario we report the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks in the right panel together with the point estimates we report the  and   percentiles figure   final size distribution left panel and mean peak incidence for scenario ias yellow scenario ibs green and scenario ibtbs blue when the quarantine contact rate is  q   r   and the proportion of asymptomatic individuals is  we consider the a reproductive number that is different for asymptomatic r a   and symptomatic r s   individuals r s    and r a     r s   in the left panel for each scenario we report the probability that a simulation leads to a number of cases smaller than the  of the population purple asterisks in the right panel together with the point estimates we report the  and   percentiles ']\n \n-----------------\nGROUP DESCRIPTIVE KEYWORDS\n model\n epidemic\n number\n population\n individuals\n preprint\n cases\n infected\n medrxiv\n time\n-----------------\n","name":"stdout"}]},{"metadata":{"trusted":true},"cell_type":"code","source":"from sklearn.metrics import precision_score, recall_score\nfrom sklearn.metrics import accuracy_score\nfrom sklearn.metrics import f1_score\n    \nprint(model_name, \":\\n\")\nprint(\"Accuracy Score: \", '{:,.3f}'.format(float(accuracy_score(y_test, y_pred)) * 100), \"%\")\nprint(\"F1 score: \", '{:,.3f}'.format(float(f1_score(test, pred, average='micro')) * 100), \"%\")","execution_count":null,"outputs":[]}],"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat":4,"nbformat_minor":4}